// @shopware-ag/meteor-admin-sdk/umd/context/index.js
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
(function (factory) {
    if (typeof module === "object" && typeof module.exports === "object") {
        var v = factory(require, exports);
        if (v !== undefined) module.exports = v;
    }
    else if (typeof define === "function" && define.amd) {
        define(["require", "exports", "../channel", "./compare-version", "./acl"], factory);
    }
})(function (require, exports) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    exports.getShopId = exports.getModuleInformation = exports.can = exports.getAppInformation = exports.getUserTimezone = exports.getUserInformation = exports.compareIsShopwareVersion = exports.getShopwareVersion = exports.getCurrency = exports.subscribeLocale = exports.getLocale = exports.getEnvironment = exports.subscribeLanguage = exports.getLanguage = void 0;
    const channel_1 = require("../channel");
    const compare_version_1 = __importDefault(require("./compare-version"));
    const acl_1 = __importDefault(require("./acl"));
    exports.getLanguage = (0, channel_1.createSender)('contextLanguage', {});
    exports.subscribeLanguage = (0, channel_1.createSubscriber)('contextLanguage');
    exports.getEnvironment = (0, channel_1.createSender)('contextEnvironment', {});
    exports.getLocale = (0, channel_1.createSender)('contextLocale', {});
    exports.subscribeLocale = (0, channel_1.createSubscriber)('contextLocale');
    exports.getCurrency = (0, channel_1.createSender)('contextCurrency', {});
    exports.getShopwareVersion = (0, channel_1.createSender)('contextShopwareVersion', {});
    exports.compareIsShopwareVersion = (0, compare_version_1.default)(exports.getShopwareVersion);
    exports.getUserInformation = (0, channel_1.createSender)('contextUserInformation', {});
    exports.getUserTimezone = (0, channel_1.createSender)('contextUserTimezone', {});
    exports.getAppInformation = (0, channel_1.createSender)('contextAppInformation', {});
    exports.can = (0, acl_1.default)(exports.getAppInformation);
    exports.getModuleInformation = (0, channel_1.createSender)('contextModuleInformation', {});
    exports.getShopId = (0, channel_1.createSender)('contextShopId', {});
});
