// @shopware-ag/meteor-component-library/dist/esm/index.js
import '../index.css';
import { default as default2, default as default3 } from "./MtDatepicker.js";
import { default as default4, default as default5 } from "./MtBanner.js";
import { default as default6 } from "./MtBadge.js";
import { default as default7, default as default8 } from "./MtLoader.js";
import { default as default9, default as default10 } from "./MtProgressBar.js";
import { default as default11, default as default12 } from "./MtButton.js";
import { default as default13 } from "./MtChart.js";
import { default as default14, default as default15 } from "./MtCheckbox.js";
import { default as default16, default as default17 } from "./MtColorpicker.js";
import { default as default18, default as default19 } from "./MtEmailField.js";
import { default as default20 } from "./MtLink.js";
import { default as default21, default as default22 } from "./MtNumberField.js";
import { default as default23, default as default24 } from "./MtPasswordField.js";
import { default as default25, default as default26 } from "./MtSelect.js";
import { default as default27 } from "./MtSlider.js";
import { _, _ as _2 } from "../mt-switch.vue_vue_type_style_index_0_lang-e05ec27a.mjs";
import { default as default28, default as default29 } from "./MtTextField.js";
import { default as default30, default as default31 } from "./MtTextarea.js";
import { _ as _3, _ as _4 } from "../mt-icon.vue_vue_type_style_index_0_lang-2cc5f73e.mjs";
import { default as default32, default as default33 } from "./MtCard.js";
import { default as default34 } from "./MtEmptyState.js";
import { default as default35, default as default36 } from "./MtTabs.js";
import { default as default37, default as default38 } from "./MtDataTable.js";
import { default as default39, default as default40 } from "./MtPagination.js";
import { default as default41, default as default42 } from "./MtSkeletonBar.js";
import { default as default43 } from "./MtToast.js";
import { m, u } from "../mt-snackbar-9786f84c.mjs";
import { default as default44 } from "./MtAvatar.js";
import { default as default45 } from "./MtPromoBadge.js";
import { default as default46 } from "./MtPopover.js";
import { default as default47 } from "./MtPopoverItem.js";
import { M } from "../mt-popover-item-result-c0214b39.mjs";
import { _ as _5 } from "../mt-floating-ui.vue_vue_type_style_index_0_lang-2d3bcbd2.mjs";
import { default as default48 } from "./MtModal.js";
import { default as default49 } from "./MtModalRoot.js";
import { _ as _6 } from "../mt-modal-close.vue_vue_type_script_setup_true_lang-15d65d8b.mjs";
import { _ as _7 } from "../mt-modal-trigger.vue_vue_type_script_setup_true_lang-60b19ac9.mjs";
import { _ as _8 } from "../mt-modal-action.vue_vue_type_script_setup_true_lang-cb0038ce.mjs";
import { default as default50 } from "./MtText.js";
import { default as default51 } from "./MtInset.js";
import { _ as _9 } from "../mt-theme-provider.vue_vue_type_script_setup_true_lang-818639ad.mjs";
import { v } from "../tooltip.directive-a5042569.mjs";
import { d as debounce } from "../debounce-b18437e2.mjs";
import { default as default52 } from "./MtTooltip.js";
import { default as default53 } from "./MtTextEditor.js";
import { default as default54 } from "./MtTextEditorToolbarButton.js";
import { default as default55 } from "./MtSearch.js";
import { default as default56 } from "./MtUrlField.js";
import { _ as _10 } from "../mt-unit-field.vue_vue_type_script_setup_true_lang-0c86124e.mjs";
import { _ as _11 } from "../mt-entity-data-table.vue_vue_type_style_index_0_lang-7afc50bc.mjs";
import { _ as _12 } from "../mt-entity-select.vue_vue_type_script_setup_true_lang-f4392f60.mjs";
import "vue";
import "./MtHelpText.js";
import "../_plugin-vue_export-helper-cc2b3d55.mjs";
import "./MtFieldLabel.js";
import "./MtInheritanceSwitch.js";
import "vue-i18n";
import "./MtFieldError.js";
import "../useFutureFlags-2be3e179.mjs";
import "../useIsInsideTooltip-0c3bd290.mjs";
import "../object-e11d5dd3.mjs";
import "../index-c0f92ea5.mjs";
import "../index-221bad05.mjs";
import "../mt-base-field-7a978dcf.mjs";
import "./MtFieldCopyable.js";
import "../id-1e5b8276.mjs";
import "../focus-trap.esm-3b050c4a.mjs";
import "./MtFieldAffix.js";
import "./MtSelectBase.js";
import "./MtSelectResultList.js";
import "./MtPopoverDeprecated.js";
import "../provideInjectKeys-47aad241.mjs";
import "./MtSelectResult.js";
import "./MtSelectSelectionList.js";
import "../mt-label.vue_vue_type_style_index_0_lang-13843d32.mjs";
import "./MtColorBadge.js";
import "../mt-highlight-text.vue_vue_type_style_index_0_lang-550f3bfe.mjs";
import "../mt-context-button.vue_vue_type_style_index_0_lang-08ff766e.mjs";
import "./MtContextMenuItem.js";
import "./MtPriorityPlusNavigation.js";
import "./MtDataTableSettings.js";
import "./MtDataTableTextRenderer.js";
import "./MtDataTableNumberRenderer.js";
import "./MtDataTableBadgeRenderer.js";
import "./MtDataTablePriceRenderer.js";
import "./MtSegmentedControl.js";
import "./MtDataTableResetFilterButton.js";
import "./MtDataTableFilter.js";
import "./MtToastNotification.js";
import "./MtSnackbarNotification.js";
import "./MtSmoothReflow.js";
import "../_commonjsHelpers-7a77ea84.mjs";
import "../floating-ui.dom-bd79e17e.mjs";
import "../useModalContext-3b7570bc.mjs";
import "../floating-ui.vue-f3249372.mjs";
import "./MtTextEditorToolbar.js";
import "./MtTextEditorToolbarButtonColor.js";
import "./MtTextEditorToolbarButtonLink.js";
import "./MtTextEditorToolbarButtonTable.js";
import "./MtTextEditorDiffModal.js";
import "./MtUnitSelect.js";
import "./MtUnitSelectResult.js";
import "../useRepository-829c0d90.mjs";
function DeviceHelper() {
  this.listeners = [];
  window.addEventListener("resize", this.resize.bind(this));
}
DeviceHelper.prototype = Object.assign(DeviceHelper.prototype, {
  /**
   * Resize method which will be fired when the user resizes the browser.
   *
   * @returns {void}
   */
  resize: debounce(function debouncedResize(event) {
    this.listeners.forEach((listenerObject) => {
      listenerObject.listener.call(listenerObject.scope, event);
    });
  }, 100),
  /**
   * Registers an event register for the browser "resize" event.
   *
   * @param {Function} callback
   * @param {Any} scope
   * @param {Object} component
   * @returns {number}
   */
  onResize({ listener, scope, component }) {
    if (!scope) {
      scope = window;
    }
    this.listeners.push({ listener, scope, component });
    return this.listeners.length - 1;
  },
  removeResizeListener(component) {
    this.listeners = this.listeners.filter(
      (listenerObject) => component !== listenerObject.component
    );
    return true;
  },
  /**
   * Returns the user agent string.
   *
   * @returns {string}
   */
  getUserAgent() {
    return window.navigator.userAgent;
  },
  /**
   * Returns the current viewport with in pixels.
   * @returns {number}
   */
  getViewportWidth() {
    return window.innerWidth;
  },
  /**
   * Returns the current viewport height in pixels.
   *
   * @returns {number}
   */
  getViewportHeight() {
    return window.innerHeight;
  },
  /**
   * Returns the pixel ratio of the device as a number.
   *
   * @returns {number}
   */
  getDevicePixelRatio() {
    return window.devicePixelRatio;
  },
  /**
   * Returns the device screen width in pixels.
   *
   * @returns {number}
   */
  getScreenWidth() {
    return window.screen.width;
  },
  /**
   * Returns the device screen height in pixels.
   *
   * @returns {number}
   */
  getScreenHeight() {
    return window.screen.height;
  },
  /**
   * Returns information about the screen orientation.
   *
   * @returns {object}
   */
  getScreenOrientation() {
    return window.screen.orientation;
  },
  /**
   * Returns the current browser language as a string.
   *
   * @returns {string}
   */
  getBrowserLanguage() {
    return window.navigator.language;
  },
  /**
   * Returns the current platform (e.g. "Win32") as a string.
   *
   * @returns {string}
   */
  getPlatform() {
    return window.navigator.platform;
  },
  /**
   * Returns the system-key (e.g. "CTRL") as a string depending of the current operating system.
   *
   * @returns {string}
   */
  getSystemKey() {
    return this.getPlatform().indexOf("Mac") > -1 ? "CTRL" : "ALT";
  }
});
const DeviceHelperPlugin = {
  install(app) {
    const deviceHelper = new DeviceHelper();
    app.config.globalProperties.$device = deviceHelper;
    app.mixin({
      unmounted() {
        this.$device.removeResizeListener(this);
      }
    });
    return true;
  }
};
const DeviceHelperPlugin$1 = DeviceHelperPlugin;
const all = "";
export {
  DeviceHelperPlugin$1 as DeviceHelperPlugin,
  default44 as MtAvatar,
  default6 as MtBadge,
  default4 as MtBanner,
  default11 as MtButton,
  default32 as MtCard,
  default13 as MtChart,
  default14 as MtCheckbox,
  default16 as MtColorpicker,
  default37 as MtDataTable,
  default2 as MtDatepicker,
  default18 as MtEmailField,
  default34 as MtEmptyState,
  _11 as MtEntityDataTable,
  _12 as MtEntitySelect,
  _5 as MtFloatingUi,
  _3 as MtIcon,
  default51 as MtInset,
  default20 as MtLink,
  default7 as MtLoader,
  default48 as MtModal,
  _8 as MtModalAction,
  _6 as MtModalClose,
  default49 as MtModalRoot,
  _7 as MtModalTrigger,
  default21 as MtNumberField,
  default39 as MtPagination,
  default23 as MtPasswordField,
  default46 as MtPopover,
  default47 as MtPopoverItem,
  M as MtPopoverItemResult,
  default9 as MtProgressBar,
  default45 as MtPromoBadge,
  default55 as MtSearch,
  default25 as MtSelect,
  default41 as MtSkeletonBar,
  default27 as MtSlider,
  m as MtSnackbar,
  _ as MtSwitch,
  default35 as MtTabs,
  default50 as MtText,
  default53 as MtTextEditor,
  default54 as MtTextEditorToolbarButton,
  default28 as MtTextField,
  default30 as MtTextarea,
  _9 as MtThemeProvider,
  default43 as MtToast,
  default52 as MtTooltip,
  _10 as MtUnitField,
  default56 as MtUrlField,
  default5 as SwBanner,
  default12 as SwButton,
  default33 as SwCard,
  default15 as SwCheckbox,
  default17 as SwColorpicker,
  default38 as SwDataTable,
  default3 as SwDatepicker,
  default19 as SwEmailField,
  _4 as SwIcon,
  default8 as SwLoader,
  default22 as SwNumberField,
  default40 as SwPagination,
  default24 as SwPasswordField,
  default10 as SwProgressBar,
  default26 as SwSelect,
  default42 as SwSkeletonBar,
  _2 as SwSwitch,
  default36 as SwTabs,
  default29 as SwTextField,
  default31 as SwTextarea,
  v as TooltipDirective,
  u as useSnackbar
};
