// @shopware-ag/meteor-component-library/dist/esm/MtDataTableTextRenderer.js
import '../mt-data-table-text-renderer.css';
import { defineComponent, computed, openBlock, createElementBlock, createCommentVNode, withModifiers, toDisplayString } from "vue";
import { a as get } from "../object-e11d5dd3.mjs";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
const _sfc_main = defineComponent({
  name: "MtDataTableTextRenderer",
  props: {
    columnDefinition: {
      type: Object,
      required: true
    },
    data: {
      type: void 0,
      required: true
    }
  },
  setup(props) {
    const renderPreviewImage = computed(() => {
      return get(props.data, props.columnDefinition.previewImage || "");
    });
    const renderString = computed(() => {
      return get(props.data, props.columnDefinition.property);
    });
    return {
      renderString,
      renderPreviewImage
    };
  }
});
const mtDataTableTextRenderer_vue_vue_type_style_index_0_scoped_745a4686_lang = "";
const _hoisted_1 = { class: "mt-data-table-text-renderer-cell" };
const _hoisted_2 = {
  key: 0,
  class: "mt-data-table-preview-image-renderer"
};
const _hoisted_3 = ["src", "alt"];
const _hoisted_4 = {
  key: 2,
  class: "mt-data-table-text-renderer"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return openBlock(), createElementBlock("div", _hoisted_1, [
    _ctx.columnDefinition.previewImage ? (openBlock(), createElementBlock("div", _hoisted_2, [
      _ctx.columnDefinition.previewImage ? (openBlock(), createElementBlock("img", {
        key: 0,
        class: "mt-data-table-preview-image-renderer-item",
        src: _ctx.renderPreviewImage,
        alt: _ctx.renderString
      }, null, 8, _hoisted_3)) : createCommentVNode("", true)
    ])) : createCommentVNode("", true),
    _ctx.columnDefinition.clickable ? (openBlock(), createElementBlock("a", {
      key: 1,
      class: "mt-data-table-text-renderer",
      href: "#",
      onClick: _cache[0] || (_cache[0] = withModifiers(($event) => _ctx.$emit("click", _ctx.data), ["prevent"]))
    }, toDisplayString(_ctx.renderString), 1)) : (openBlock(), createElementBlock("p", _hoisted_4, toDisplayString(_ctx.renderString), 1))
  ]);
}
const MtDataTableTextRenderer = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render], ["__scopeId", "data-v-745a4686"]]);
export {
  MtDataTableTextRenderer as default
};
