// @shopware-ag/meteor-component-library/dist/esm/MtDataTableNumberRenderer.js
import '../mt-data-table-number-renderer.css';
import { defineComponent, computed, openBlock, createElementBlock, withModifiers, toDisplayString } from "vue";
import { a as get } from "../object-e11d5dd3.mjs";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
const _sfc_main = defineComponent({
  name: "MtDataTableNumberRenderer",
  props: {
    columnDefinition: {
      type: Object,
      required: true
    },
    data: {
      type: void 0,
      required: true
    }
  },
  setup(props) {
    const renderString = computed(() => {
      return String(Number(get(props.data, props.columnDefinition.property)));
    });
    return {
      renderString
    };
  }
});
const mtDataTableNumberRenderer_vue_vue_type_style_index_0_lang = "";
const _hoisted_1 = {
  key: 1,
  class: "mt-data-table-number-renderer"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return _ctx.columnDefinition.clickable ? (openBlock(), createElementBlock("a", {
    key: 0,
    class: "mt-data-table-number-renderer",
    href: "#",
    onClick: _cache[0] || (_cache[0] = withModifiers(($event) => _ctx.$emit("click", _ctx.data), ["prevent"]))
  }, toDisplayString(_ctx.renderString), 1)) : (openBlock(), createElementBlock("p", _hoisted_1, toDisplayString(_ctx.renderString), 1));
}
const MtDataTableNumberRenderer = /* @__PURE__ */ _export_sfc(_sfc_main, [["render", _sfc_render]]);
export {
  MtDataTableNumberRenderer as default
};
