// @shopware-ag/meteor-component-library/dist/esm/MtDataTableResetFilterButton.js
import '../mt-data-table-reset-filter-button.css';
import { defineComponent, openBlock, createElementBlock, toDisplayString, unref } from "vue";
import { useI18n } from "vue-i18n";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
const _hoisted_1 = {
  type: "button",
  class: "mt-data-table-reset-filter-button"
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-data-table-reset-filter-button",
  props: {
    numberOfAppliedFilters: {}
  },
  setup(__props) {
    const { t } = useI18n({
      messages: {
        en: {
          label: "Remove filter | Remove filters"
        },
        de: {
          label: "Filter entfernen | Filter entfernen"
        }
      }
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("button", _hoisted_1, toDisplayString(unref(t)("label", { n: _ctx.numberOfAppliedFilters })), 1);
    };
  }
});
const mtDataTableResetFilterButton_vue_vue_type_style_index_0_scoped_ea9bb72d_lang = "";
const MtDataTableResetFilterButton = /* @__PURE__ */ _export_sfc(_sfc_main, [["__scopeId", "data-v-ea9bb72d"]]);
export {
  MtDataTableResetFilterButton as default
};
