// @shopware-ag/meteor-component-library/dist/esm/MtDataTableFilter.js
import '../mt-data-table-filter.css';
import { defineComponent, openBlock, createElementBlock, createVNode, withCtx, createTextVNode, toDisplayString, unref, createElementVNode, withKeys, Fragment, renderList, createBlock } from "vue";
import { _ as _sfc_main$1 } from "../mt-icon.vue_vue_type_style_index_0_lang-2cc5f73e.mjs";
import mtPopover from "./MtPopover.js";
import mtPopoverItem from "./MtPopoverItem.js";
import MtText from "./MtText.js";
import { useI18n } from "vue-i18n";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
import "./MtCheckbox.js";
import "../mt-base-field-7a978dcf.mjs";
import "./MtInheritanceSwitch.js";
import "./MtTooltip.js";
import "../floating-ui.vue-f3249372.mjs";
import "../floating-ui.dom-bd79e17e.mjs";
import "../useIsInsideTooltip-0c3bd290.mjs";
import "../index-221bad05.mjs";
import "./MtFieldCopyable.js";
import "../tooltip.directive-a5042569.mjs";
import "../id-1e5b8276.mjs";
import "./MtHelpText.js";
import "../useFutureFlags-2be3e179.mjs";
import "./MtFieldError.js";
import "../mt-switch.vue_vue_type_style_index_0_lang-e05ec27a.mjs";
import "./MtFieldLabel.js";
import "./MtButton.js";
import "./MtLoader.js";
import "./MtSmoothReflow.js";
import "../_commonjsHelpers-7a77ea84.mjs";
import "../mt-floating-ui.vue_vue_type_style_index_0_lang-2d3bcbd2.mjs";
const _hoisted_1 = {
  class: "mt-data-table-filter",
  "data-testid": "mt-data-table-filter"
};
const _hoisted_2 = ["onClick"];
const _hoisted_3 = ["aria-label"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-data-table-filter",
  props: {
    filter: {},
    appliedOptions: {}
  },
  emits: ["removeOption", "addOption", "removeFilter"],
  setup(__props) {
    const props = __props;
    function isOptionSelected(optionId) {
      return !!props.appliedOptions.find((option) => option.id === optionId);
    }
    const { t } = useI18n({
      messages: {
        en: {
          is: "is",
          removeButton: "Remove filter"
        },
        de: {
          is: "ist",
          removeButton: "Filter entfernen"
        }
      }
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        createVNode(MtText, {
          class: "mt-data-table-filter__property",
          size: "2xs"
        }, {
          default: withCtx(() => [
            createTextVNode(toDisplayString(_ctx.filter.label), 1)
          ]),
          _: 1
        }),
        createVNode(MtText, {
          size: "2xs",
          color: "color-text-secondary-default",
          class: "mt-data-table-filter__rule"
        }, {
          default: withCtx(() => [
            createTextVNode(toDisplayString(unref(t)("is")), 1)
          ]),
          _: 1
        }),
        createVNode(mtPopover, {
          class: "mt-data-table-filter__option",
          title: "Manufactuer"
        }, {
          trigger: withCtx(({ toggleFloatingUi }) => [
            createElementVNode("button", {
              type: "button",
              onKeydown: _cache[0] || (_cache[0] = withKeys(($event) => _ctx.$emit("removeFilter"), ["delete"])),
              onClick: toggleFloatingUi
            }, toDisplayString(_ctx.appliedOptions.map((option) => option.label).join(", ")), 41, _hoisted_2)
          ]),
          "popover-items__base": withCtx(() => [
            (openBlock(true), createElementBlock(Fragment, null, renderList(_ctx.filter.type.options, (option) => {
              return openBlock(), createBlock(mtPopoverItem, {
                key: option.id,
                label: option.label,
                "show-checkbox": "",
                "checkbox-checked": isOptionSelected(option.id),
                onChangeCheckbox: (isChecked) => isChecked ? _ctx.$emit("addOption", _ctx.filter.id, option.id) : _ctx.$emit("removeOption", _ctx.filter.id, option.id)
              }, null, 8, ["label", "checkbox-checked", "onChangeCheckbox"]);
            }), 128))
          ]),
          _: 1
        }),
        createElementVNode("button", {
          type: "button",
          class: "mt-data-table-filter__remove-button",
          "aria-label": unref(t)("removeButton"),
          onKeydown: _cache[1] || (_cache[1] = withKeys(($event) => _ctx.$emit("removeFilter"), ["delete"])),
          onClick: _cache[2] || (_cache[2] = ($event) => _ctx.$emit("removeFilter"))
        }, [
          createVNode(_sfc_main$1, {
            name: "solid-times-xxs",
            "aria-hidden": "true"
          })
        ], 40, _hoisted_3)
      ]);
    };
  }
});
const mtDataTableFilter_vue_vue_type_style_index_0_scoped_afdb8a25_lang = "";
const MtDataTableFilter = /* @__PURE__ */ _export_sfc(_sfc_main, [["__scopeId", "data-v-afdb8a25"]]);
export {
  MtDataTableFilter as default
};
