// @shopware-ag/meteor-component-library/dist/esm/MtHelpText.js
import '../mt-help-text.css';
import { defineComponent, openBlock, createBlock, withCtx, createElementVNode, mergeProps, createVNode } from "vue";
import { _ as _sfc_main$1 } from "../mt-icon.vue_vue_type_style_index_0_lang-2cc5f73e.mjs";
import MtTooltip from "./MtTooltip.js";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
import "../floating-ui.vue-f3249372.mjs";
import "../floating-ui.dom-bd79e17e.mjs";
import "../useIsInsideTooltip-0c3bd290.mjs";
import "../index-221bad05.mjs";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-help-text",
  props: {
    text: {},
    width: {},
    showDelay: { default: 0 },
    hideDelay: { default: 150 },
    placement: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(MtTooltip, {
        content: _ctx.text,
        placement: _ctx.placement,
        "max-width": _ctx.width,
        "delay-duration-in-ms": _ctx.showDelay,
        "hide-delay-duration-in-ms": _ctx.hideDelay
      }, {
        default: withCtx((props) => [
          createElementVNode("button", mergeProps({ type: "button" }, { ...props, ..._ctx.$attrs }, { class: "mt-help-text" }), [
            createVNode(_sfc_main$1, {
              "aria-hidden": "true",
              "data-testid": "mt-help-text__icon",
              name: "solid-question-circle-s"
            })
          ], 16)
        ]),
        _: 1
      }, 8, ["content", "placement", "max-width", "delay-duration-in-ms", "hide-delay-duration-in-ms"]);
    };
  }
});
const mtHelpText_vue_vue_type_style_index_0_scoped_9f6b1469_lang = "";
const MtHelpText = /* @__PURE__ */ _export_sfc(_sfc_main, [["__scopeId", "data-v-9f6b1469"]]);
export {
  MtHelpText as default
};
