// @shopware-ag/meteor-component-library/dist/esm/MtTextEditorToolbarButtonColor.js
import '../mt-text-editor-toolbar-button-color.css';
import { defineComponent, openBlock, createElementBlock, createVNode } from "vue";
import MtColorpicker from "./MtColorpicker.js";
import { useI18n } from "vue-i18n";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
import "../debounce-b18437e2.mjs";
import "../mt-base-field-7a978dcf.mjs";
import "./MtInheritanceSwitch.js";
import "../mt-icon.vue_vue_type_style_index_0_lang-2cc5f73e.mjs";
import "./MtTooltip.js";
import "../floating-ui.vue-f3249372.mjs";
import "../floating-ui.dom-bd79e17e.mjs";
import "../useIsInsideTooltip-0c3bd290.mjs";
import "../index-221bad05.mjs";
import "./MtFieldCopyable.js";
import "../tooltip.directive-a5042569.mjs";
import "../id-1e5b8276.mjs";
import "./MtHelpText.js";
import "../useFutureFlags-2be3e179.mjs";
import "../mt-floating-ui.vue_vue_type_style_index_0_lang-2d3bcbd2.mjs";
import "./MtText.js";
import "../focus-trap.esm-3b050c4a.mjs";
import "./MtButton.js";
import "./MtLoader.js";
import "./MtFieldError.js";
const _hoisted_1 = { class: "mt-text-editor-toolbar-button-color" };
const colorButton = {
  name: "text-color",
  label: "mt-text-editor-toolbar-button-color.label",
  position: 2e3
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-text-editor-toolbar-button-color",
  props: {
    editor: {
      type: Object,
      required: true
    },
    button: {
      type: Object,
      required: true
    },
    disabled: {
      type: Boolean,
      default: false
    }
  },
  setup(__props) {
    useI18n({
      useScope: "global",
      messages: {
        en: {
          "mt-text-editor-toolbar-button-color": {
            label: "Text Color"
          }
        },
        de: {
          "mt-text-editor-toolbar-button-color": {
            label: "Textfarbe"
          }
        }
      }
    });
    const props = __props;
    const getTextColor = () => {
      return props.editor.getAttributes("textStyle").color;
    };
    const onUpdateModelValue = (color) => {
      props.editor.chain().focus().setColor(color).run();
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", _hoisted_1, [
        createVNode(MtColorpicker, {
          class: "mt-text-editor-toolbar-button-color__colorpicker",
          compact: "",
          "model-value": getTextColor(),
          "onUpdate:modelValue": onUpdateModelValue,
          disabled: __props.button.disabled ? __props.button.disabled(props.editor, __props.disabled) : __props.disabled,
          "apply-mode": ""
        }, null, 8, ["model-value", "disabled"])
      ]);
    };
  }
});
const mtTextEditorToolbarButtonColor_vue_vue_type_style_index_0_scoped_356322ec_lang = "";
const mtTextEditorToolbarButtonColor = /* @__PURE__ */ _export_sfc(_sfc_main, [["__scopeId", "data-v-356322ec"]]);
export {
  colorButton,
  mtTextEditorToolbarButtonColor as default
};
