// @shopware-ag/meteor-component-library/dist/esm/MtTextEditorToolbarButtonTable.js
import '../mt-text-editor-toolbar-button-table.css';
import { defineComponent, ref, openBlock, createElementBlock, Fragment, createVNode, withCtx, unref, createElementVNode, createTextVNode, toDisplayString } from "vue";
import MtModalRoot from "./MtModalRoot.js";
import MtModal from "./MtModal.js";
import MtButton from "./MtButton.js";
import { _ as _sfc_main$1 } from "../mt-switch.vue_vue_type_style_index_0_lang-e05ec27a.mjs";
import mtTextEditorToolbarButton from "./MtTextEditorToolbarButton.js";
import MtNumberField from "./MtNumberField.js";
import { _ as _sfc_main$2 } from "../mt-modal-close.vue_vue_type_script_setup_true_lang-15d65d8b.mjs";
import { useI18n } from "vue-i18n";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
import "../useModalContext-3b7570bc.mjs";
import "../mt-icon.vue_vue_type_style_index_0_lang-2cc5f73e.mjs";
import "./MtText.js";
import "../id-1e5b8276.mjs";
import "../focus-trap.esm-3b050c4a.mjs";
import "../useIsInsideTooltip-0c3bd290.mjs";
import "./MtLoader.js";
import "./MtFieldError.js";
import "./MtFieldLabel.js";
import "./MtInheritanceSwitch.js";
import "./MtTooltip.js";
import "../floating-ui.vue-f3249372.mjs";
import "../floating-ui.dom-bd79e17e.mjs";
import "../index-221bad05.mjs";
import "./MtHelpText.js";
import "../useFutureFlags-2be3e179.mjs";
import "../tooltip.directive-a5042569.mjs";
import "./MtTextField.js";
import "../mt-base-field-7a978dcf.mjs";
import "./MtFieldCopyable.js";
const _hoisted_1 = { class: "mt-text-editor__table-modal" };
const _hoisted_2 = { class: "mt-text-editor__table-modal-footer" };
const tableButton = {
  name: "table",
  label: "mt-text-editor-toolbar-button-table.label",
  icon: "regular-table-xs",
  disabled: (editor, globalDisabled) => {
    if (globalDisabled) {
      return true;
    }
    return editor.isActive("table");
  },
  position: 13e3,
  contextualButtons: (editor) => {
    if (!editor.isActive("table")) {
      return [];
    }
    return [
      {
        name: "table-insert-row-before",
        label: "Insert row before",
        icon: "regular-insert-row-before",
        action: (editor2) => editor2.chain().focus().addRowBefore().run()
      },
      {
        name: "table-insert-row-after",
        label: "Insert row after",
        icon: "regular-insert-row-after",
        action: (editor2) => editor2.chain().focus().addRowAfter().run()
      },
      {
        name: "table-delete-row",
        label: "Delete row",
        icon: "regular-delete-row",
        action: (editor2) => editor2.chain().focus().deleteRow().run()
      },
      {
        name: "table-insert-column-before",
        label: "Insert column before",
        icon: "regular-insert-column-before",
        action: (editor2) => editor2.chain().focus().addColumnBefore().run()
      },
      {
        name: "table-insert-column-after",
        label: "Insert column after",
        icon: "regular-insert-column-after",
        action: (editor2) => editor2.chain().focus().addColumnAfter().run()
      },
      {
        name: "table-delete-column",
        label: "Delete column",
        icon: "regular-delete-column",
        action: (editor2) => editor2.chain().focus().deleteColumn().run()
      },
      {
        name: "table-remove-table",
        label: "Remove table",
        icon: "regular-times-xs",
        action: (editor2) => editor2.chain().focus().deleteTable().run()
      }
    ];
  }
};
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-text-editor-toolbar-button-table",
  props: {
    editor: {
      type: Object,
      required: true
    },
    button: {
      type: Object,
      required: true
    },
    disabled: {
      type: Boolean,
      default: false
    }
  },
  setup(__props) {
    const { t } = useI18n({
      useScope: "global",
      messages: {
        en: {
          "mt-text-editor-toolbar-button-table": {
            label: "Table",
            modalTitle: "Insert/Edit Table",
            insertTable: "Insert table",
            cancel: "Cancel",
            columns: "Columns",
            rows: "Rows",
            showHeader: "Show header",
            insertRowBefore: "Insert row before",
            insertRowAfter: "Insert row after",
            deleteRow: "Delete row",
            insertColumnBefore: "Insert column before",
            insertColumnAfter: "Insert column after",
            deleteColumn: "Delete column",
            removeTable: "Remove table"
          }
        },
        de: {
          "mt-text-editor-toolbar-button-table": {
            label: "Tabelle",
            modalTitle: "Tabelle einfügen/bearbeiten",
            insertTable: "Tabelle einfügen",
            cancel: "Abbrechen",
            columns: "Spalten",
            rows: "Zeilen",
            showHeader: "Kopfzeile anzeigen",
            insertRowBefore: "Zeile davor einfügen",
            insertRowAfter: "Zeile danach einfügen",
            deleteRow: "Zeile löschen",
            insertColumnBefore: "Spalte davor einfügen",
            insertColumnAfter: "Spalte danach einfügen",
            deleteColumn: "Spalte löschen",
            removeTable: "Tabelle entfernen"
          }
        }
      }
    });
    const props = __props;
    const showTableModal = ref(false);
    const tableColumns = ref(3);
    const tableRows = ref(3);
    const tableShowHeader = ref(true);
    const openTableModal = () => {
      if (props.editor.isActive("table")) {
        return;
      }
      tableColumns.value = 3;
      tableRows.value = 3;
      tableShowHeader.value = true;
      showTableModal.value = true;
    };
    const insertTable = () => {
      props.editor.chain().focus().insertTable({
        rows: tableRows.value,
        cols: tableColumns.value,
        withHeaderRow: tableShowHeader.value
      }).run();
      showTableModal.value = false;
    };
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock(Fragment, null, [
        createVNode(mtTextEditorToolbarButton, {
          button: __props.button,
          editor: props.editor,
          disabled: __props.disabled,
          onClick: openTableModal
        }, null, 8, ["button", "editor", "disabled"]),
        createVNode(MtModalRoot, {
          isOpen: showTableModal.value,
          onChange: _cache[3] || (_cache[3] = ($event) => showTableModal.value = $event)
        }, {
          default: withCtx(() => [
            createVNode(MtModal, {
              title: unref(t)("mt-text-editor-toolbar-button-table.modalTitle"),
              width: "s"
            }, {
              default: withCtx(() => [
                createElementVNode("div", _hoisted_1, [
                  createVNode(MtNumberField, {
                    label: unref(t)("mt-text-editor-toolbar-button-table.columns"),
                    modelValue: tableColumns.value,
                    "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => tableColumns.value = $event),
                    type: "number",
                    min: 1,
                    numberType: "int"
                  }, null, 8, ["label", "modelValue"]),
                  createVNode(MtNumberField, {
                    label: unref(t)("mt-text-editor-toolbar-button-table.rows"),
                    modelValue: tableRows.value,
                    "onUpdate:modelValue": _cache[1] || (_cache[1] = ($event) => tableRows.value = $event),
                    type: "number",
                    min: 1,
                    numberType: "int"
                  }, null, 8, ["label", "modelValue"]),
                  createVNode(_sfc_main$1, {
                    label: unref(t)("mt-text-editor-toolbar-button-table.showHeader"),
                    modelValue: tableShowHeader.value,
                    "onUpdate:modelValue": _cache[2] || (_cache[2] = ($event) => tableShowHeader.value = $event)
                  }, null, 8, ["label", "modelValue"])
                ])
              ]),
              footer: withCtx(() => [
                createElementVNode("div", _hoisted_2, [
                  createVNode(_sfc_main$2, {
                    as: MtButton,
                    variant: "secondary"
                  }, {
                    default: withCtx(() => [
                      createTextVNode(toDisplayString(unref(t)("mt-text-editor-toolbar-button-table.cancel")), 1)
                    ]),
                    _: 1
                  }),
                  createVNode(MtButton, {
                    variant: "primary",
                    onClick: insertTable
                  }, {
                    default: withCtx(() => [
                      createTextVNode(toDisplayString(unref(t)("mt-text-editor-toolbar-button-table.insertTable")), 1)
                    ]),
                    _: 1
                  })
                ])
              ]),
              _: 1
            }, 8, ["title"])
          ]),
          _: 1
        }, 8, ["isOpen"])
      ], 64);
    };
  }
});
const mtTextEditorToolbarButtonTable_vue_vue_type_style_index_0_scoped_33d27363_lang = "";
const mtTextEditorToolbarButtonTable = /* @__PURE__ */ _export_sfc(_sfc_main, [["__scopeId", "data-v-33d27363"]]);
export {
  mtTextEditorToolbarButtonTable as default,
  tableButton
};
