// @shopware-ag/meteor-component-library/dist/esm/MtText.js
import '../mt-text.css';
import { defineComponent, openBlock, createBlock, resolveDynamicComponent, normalizeClass, normalizeStyle, withCtx, renderSlot } from "vue";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-text",
  props: {
    size: { default: "s" },
    weight: { default: "regular" },
    color: { default: "color-text-primary-default" },
    as: { default: "p" }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(resolveDynamicComponent(_ctx.as), {
        class: normalizeClass([`mt-text--size-${_ctx.size}`, `mt-text--weight-${_ctx.weight}`]),
        style: normalizeStyle({ color: `var(--${_ctx.color})` })
      }, {
        default: withCtx(() => [
          renderSlot(_ctx.$slots, "default", {}, void 0, true)
        ]),
        _: 3
      }, 8, ["class", "style"]);
    };
  }
});
const mtText_vue_vue_type_style_index_0_scoped_44e2f2b3_lang = "";
const MtText = /* @__PURE__ */ _export_sfc(_sfc_main, [["__scopeId", "data-v-44e2f2b3"]]);
export {
  MtText as default
};
