// @shopware-ag/meteor-component-library/dist/esm/MtUnitSelectResult.js
import '../mt-unit-select-result.css';
import { defineComponent, openBlock, createElementBlock, createElementVNode, toDisplayString } from "vue";
import { _ as _export_sfc } from "../_plugin-vue_export-helper-cc2b3d55.mjs";
const _hoisted_1 = {
  role: "option",
  "aria-selected": false
};
const _hoisted_2 = ["aria-label", "data-testid"];
const _hoisted_3 = { class: "mt-unit-select-result__label" };
const _hoisted_4 = { class: "mt-unit-select-result__unit" };
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-unit-select-result",
  props: {
    unit: {},
    unitLabel: {}
  },
  emits: ["click"],
  setup(__props, { emit: __emit }) {
    const emit = __emit;
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("li", _hoisted_1, [
        createElementVNode("button", {
          class: "mt-unit-select-result",
          onClick: _cache[0] || (_cache[0] = ($event) => emit("click", _ctx.unit)),
          type: "button",
          "aria-label": `Select ${_ctx.unitLabel.plural} (${_ctx.unit})`,
          "data-testid": `unit-select-option-${_ctx.unit}`
        }, [
          createElementVNode("span", _hoisted_3, toDisplayString(_ctx.unitLabel.plural), 1),
          createElementVNode("span", _hoisted_4, toDisplayString(_ctx.unit), 1)
        ], 8, _hoisted_2)
      ]);
    };
  }
});
const mtUnitSelectResult_vue_vue_type_style_index_0_scoped_bbacdfea_lang = "";
const MtUnitSelectResult = /* @__PURE__ */ _export_sfc(_sfc_main, [["__scopeId", "data-v-bbacdfea"]]);
export {
  MtUnitSelectResult as default
};
