// @shopware-ag/meteor-component-library/dist/mt-context-button.vue_vue_type_style_index_0_lang-08ff766e.mjs
import './mt-context-button.css';
import { defineComponent, openBlock, createBlock, withCtx, renderSlot, createElementVNode, createVNode } from "vue";
import { _ as _sfc_main$1 } from "./mt-icon.vue_vue_type_style_index_0_lang-2cc5f73e.mjs";
import mtPopover from "./esm/MtPopover.js";
const _hoisted_1 = ["onClick"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-context-button",
  props: {
    menuWidth: { default: 220 },
    menuHorizontalAlign: { default: "right" },
    menuVerticalAlign: { default: "bottom" },
    icon: { default: "solid-ellipsis-h-s" },
    disabled: { type: Boolean },
    hasError: { type: Boolean },
    autoClose: { type: Boolean },
    title: { default: "" },
    childViews: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return openBlock(), createBlock(mtPopover, {
        class: "mt-context-button",
        title: _ctx.title,
        "child-views": _ctx.childViews
      }, {
        trigger: withCtx(({ toggleFloatingUi }) => [
          renderSlot(_ctx.$slots, "button", {}, () => [
            createElementVNode("button", {
              type: "button",
              role: "button",
              "aria-haspopup": "menu",
              "aria-label": "Context menu",
              class: "mt-context-button__button",
              onClick: toggleFloatingUi
            }, [
              createVNode(_sfc_main$1, {
                name: _ctx.icon,
                small: "",
                decorative: ""
              }, null, 8, ["name"]),
              renderSlot(_ctx.$slots, "button-text")
            ], 8, _hoisted_1)
          ])
        ]),
        "popover-items__base": withCtx(({ toggleFloatingUi, changeView }) => [
          renderSlot(_ctx.$slots, "default", {
            changeView,
            toggleFloatingUi
          })
        ]),
        _: 3
      }, 8, ["title", "child-views"]);
    };
  }
});
const mtContextButton_vue_vue_type_style_index_0_lang = "";
export {
  _sfc_main as _
};
