// @shopware-ag/meteor-component-library/dist/mt-context-button.vue_vue_type_style_index_0_lang-8afa3ed1.js
import './mt-context-button.css';
"use strict";
const vue = require("vue");
const mtIcon_vue_vue_type_style_index_0_lang = require("./mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
const MtPopover = require("./common/MtPopover.js");
const _hoisted_1 = ["onClick"];
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-context-button",
  props: {
    menuWidth: { default: 220 },
    menuHorizontalAlign: { default: "right" },
    menuVerticalAlign: { default: "bottom" },
    icon: { default: "solid-ellipsis-h-s" },
    disabled: { type: Boolean },
    hasError: { type: Boolean },
    autoClose: { type: Boolean },
    title: { default: "" },
    childViews: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createBlock(MtPopover, {
        class: "mt-context-button",
        title: _ctx.title,
        "child-views": _ctx.childViews
      }, {
        trigger: vue.withCtx(({ toggleFloatingUi }) => [
          vue.renderSlot(_ctx.$slots, "button", {}, () => [
            vue.createElementVNode("button", {
              type: "button",
              role: "button",
              "aria-haspopup": "menu",
              "aria-label": "Context menu",
              class: "mt-context-button__button",
              onClick: toggleFloatingUi
            }, [
              vue.createVNode(mtIcon_vue_vue_type_style_index_0_lang._sfc_main, {
                name: _ctx.icon,
                small: "",
                decorative: ""
              }, null, 8, ["name"]),
              vue.renderSlot(_ctx.$slots, "button-text")
            ], 8, _hoisted_1)
          ])
        ]),
        "popover-items__base": vue.withCtx(({ toggleFloatingUi, changeView }) => [
          vue.renderSlot(_ctx.$slots, "default", {
            changeView,
            toggleFloatingUi
          })
        ]),
        _: 3
      }, 8, ["title", "child-views"]);
    };
  }
});
const mtContextButton_vue_vue_type_style_index_0_lang = "";
exports._sfc_main = _sfc_main;
