// @shopware-ag/meteor-component-library/dist/mt-entity-select.vue_vue_type_script_setup_true_lang-772811ef.js
"use strict";
const vue = require("vue");
const MtSelect = require("./common/MtSelect.js");
const useRepository = require("./useRepository-ddf4b97b.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-entity-select",
  props: {
    entity: {},
    repository: { default: void 0 },
    modelValue: { default: null },
    labelProperty: { default: "name" },
    valueProperty: { default: "id" },
    disabled: { type: Boolean, default: false },
    enableMultiSelection: { type: Boolean, default: false }
  },
  emits: ["update:modelValue"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const options = vue.ref([]);
    const isLoading = vue.ref(false);
    const searchTerm = vue.ref("");
    const currentPage = vue.ref(1);
    const limit = vue.ref(50);
    const total = vue.ref(0);
    const repo = useRepository.useRepository(props.entity, props.repository);
    const getPage = async (page, term) => {
      const criteria = new useRepository.Criteria(page, limit.value);
      const currentSearchTerm = term || searchTerm.value;
      if (currentSearchTerm) {
        criteria.setTerm(currentSearchTerm);
      }
      if (typeof props.labelProperty === "string") {
        criteria.addSorting({
          field: props.labelProperty,
          order: "ASC",
          naturalSorting: false
        });
      }
      try {
        const result = await repo.value.search(criteria);
        if (result) {
          total.value = result.total ?? 0;
          return [...result];
        }
      } catch (e) {
        console.error(`Could not fetch data for entity ${props.entity}`, e);
      }
      return [];
    };
    const fetchData = async (append = false, term, disableOptionsUpdate = false) => {
      isLoading.value = true;
      const newOptions = await getPage(currentPage.value, term);
      isLoading.value = false;
      if (append && !disableOptionsUpdate) {
        newOptions.forEach((newOption) => {
          const index = options.value.findIndex(
            (option) => (
              // @ts-expect-error - The value property is not typed exactly the same
              option[props.valueProperty] === newOption[props.valueProperty]
            )
          );
          if (index !== -1) {
            options.value.splice(index, 1);
          }
          options.value.push(newOption);
        });
      } else if (!disableOptionsUpdate) {
        options.value = newOptions;
      }
      return newOptions;
    };
    const fetchSelectedEntities = async () => {
      const ids = (Array.isArray(props.modelValue) ? props.modelValue : [props.modelValue]).filter(
        (v) => !!v
      );
      if (ids.length === 0) {
        return [];
      }
      const criteria = new useRepository.Criteria(1, ids.length);
      criteria.addFilter(useRepository.Criteria.equalsAny("id", ids));
      try {
        const result = await repo.value.search(criteria);
        return result ? [...result] : [];
      } catch (e) {
        console.error("Could not fetch selected entities", e);
        return [];
      }
    };
    vue.onMounted(async () => {
      isLoading.value = true;
      const pagePromise = getPage(currentPage.value);
      let selectedPromise = Promise.resolve([]);
      const hasValue = Array.isArray(props.modelValue) && props.modelValue.length > 0 || !Array.isArray(props.modelValue) && props.modelValue !== null && props.modelValue !== void 0;
      if (hasValue) {
        selectedPromise = fetchSelectedEntities();
      }
      const [pageEntities, selectedEntities] = await Promise.all([pagePromise, selectedPromise]);
      const combined = [...selectedEntities, ...pageEntities];
      const unique = combined.reduce((acc, current) => {
        if (!acc.find((item) => item.id === current.id)) {
          acc.push(current);
        }
        return acc;
      }, []);
      options.value = unique;
      isLoading.value = false;
    });
    const onItemAdd = (item) => {
      if (!props.valueProperty) {
        return;
      }
      const itemExists = options.value.some(
        (option) => option[props.valueProperty] === item[props.valueProperty]
      );
      if (!itemExists) {
        options.value.push(item);
      }
    };
    const fetchSearchData = async ({ searchTerm: newSearchTerm }) => {
      currentPage.value = 1;
      searchTerm.value = newSearchTerm;
      return fetchData(false, newSearchTerm, true);
    };
    const fetchNextPage = () => {
      if (isLoading.value) {
        return;
      }
      if (options.value.length >= total.value && total.value > 0) {
        return;
      }
      currentPage.value += 1;
      fetchData(true);
    };
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createBlock(MtSelect, vue.mergeProps(_ctx.$attrs, {
        "model-value": props.modelValue,
        options: options.value,
        "is-loading": isLoading.value,
        "label-property": props.labelProperty,
        "value-property": props.valueProperty,
        "enable-multi-selection": props.enableMultiSelection,
        disabled: props.disabled,
        "search-function": fetchSearchData,
        "onUpdate:modelValue": _cache[0] || (_cache[0] = ($event) => emit("update:modelValue", $event)),
        onPaginate: fetchNextPage,
        onItemAdd
      }), vue.createSlots({ _: 2 }, [
        vue.renderList(_ctx.$slots, (_, name) => {
          return {
            name,
            fn: vue.withCtx((scope) => [
              vue.renderSlot(_ctx.$slots, name, vue.normalizeProps(vue.guardReactiveProps(scope ?? {})))
            ])
          };
        })
      ]), 1040, ["model-value", "options", "is-loading", "label-property", "value-property", "enable-multi-selection", "disabled"]);
    };
  }
});
exports._sfc_main = _sfc_main;
