// @shopware-ag/meteor-component-library/dist/mt-entity-data-table.vue_vue_type_style_index_0_lang-4644129f.js
import './mt-entity-data-table.css';
"use strict";
const vue = require("vue");
const useRepository = require("./useRepository-ddf4b97b.js");
const MtDataTable = require("./common/MtDataTable.js");
const index = require("./index-d39b6c26.js");
const vueI18n = require("vue-i18n");
const MtModalRoot = require("./common/MtModalRoot.js");
const MtModal = require("./common/MtModal.js");
const MtButton = require("./common/MtButton.js");
const MtLoader = require("./common/MtLoader.js");
const _hoisted_1 = { class: "mt-entity-data-table__delete-modal-footer" };
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-entity-data-table",
  props: {
    entity: {},
    repository: {},
    forceRealModal: { type: Boolean },
    columns: {},
    columnChanges: {},
    title: {},
    subtitle: {},
    layout: {},
    allowBulkDelete: { type: Boolean },
    allowBulkEdit: { type: Boolean },
    allowRowSelection: { type: Boolean },
    bulkEditMoreActions: {},
    disableDelete: { type: Boolean },
    disableEdit: { type: Boolean },
    disableSearch: { type: Boolean },
    disableSettingsTable: { type: Boolean },
    additionalContextButtons: {},
    caption: {},
    paginationOptions: {},
    availableFilters: {}
  },
  emits: ["open-details", "bulk-delete", "bulk-edit"],
  setup(__props, { emit: __emit }) {
    const props = __props;
    const emit = __emit;
    const { t } = vueI18n.useI18n({
      messages: {
        en: {
          booleanFilter: {
            true: "Active",
            false: "Inactive"
          },
          entity: {
            delete: {
              title: "Delete item",
              description: "Are you sure you want to delete the selected items?",
              cancel: "Cancel",
              delete: "Delete"
            }
          }
        },
        de: {
          booleanFilter: {
            true: "Aktiv",
            false: "Inaktiv"
          },
          entity: {
            delete: {
              title: "Element löschen",
              description: "Soll dieses Element wirklich gelöscht werden?",
              cancel: "Abbrechen",
              delete: "Löschen"
            }
          }
        }
      }
    });
    const dataSource = vue.ref([]);
    const currentPage = vue.ref(1);
    const paginationLimit = vue.ref(25);
    const paginationTotalItems = vue.ref(0);
    const sortBy = vue.ref("");
    const sortDirection = vue.ref("ASC");
    const searchValue = vue.ref("");
    const isLoading = vue.ref(true);
    const enableRowNumbering = vue.ref(false);
    const showOutlines = vue.ref(true);
    const showStripes = vue.ref(true);
    const enableOutlineFraming = vue.ref(false);
    const selectedRows = vue.ref([]);
    const appliedFilters = vue.ref([]);
    const numberOfResults = vue.ref(0);
    const isDeleteModalOpen = vue.ref(false);
    const itemToDelete = vue.ref(null);
    const isDeletingEntity = vue.ref(false);
    const repository = useRepository.useRepository(props.entity, props.repository);
    const generatedFilters = index.computedAsync(async () => {
      var _a;
      const filterPromises = (_a = props.availableFilters) == null ? void 0 : _a.map(async (filter) => {
        if (filter.filterType === "boolean") {
          return {
            id: filter.id,
            label: filter.label,
            type: {
              id: "options",
              options: [
                {
                  id: "true",
                  label: t("booleanFilter.true")
                },
                {
                  id: "false",
                  label: t("booleanFilter.false")
                }
              ]
            }
          };
        }
        const repository2 = await useRepository.getRepository(
          filter.id,
          props.repository
        );
        const optionCriteria = new useRepository.Criteria(1, 500);
        optionCriteria.addIncludes({
          [filter.id]: ["id", "name"]
        });
        const options = await repository2.search(optionCriteria);
        return {
          id: filter.id,
          label: filter.label,
          type: {
            id: "options",
            options: options == null ? void 0 : options.map((option) => ({
              id: option.id,
              // @ts-expect-error - name will be used for label
              label: (option == null ? void 0 : option.name) ?? ""
            }))
          }
        };
      });
      const filters = await Promise.all(filterPromises);
      return filters;
    }, []);
    function addDataTableFilters(criteria, filters) {
      filters.forEach((filter) => {
        filter.type.options.forEach((option) => {
          criteria.addFilter(useRepository.Criteria.equals(filter.id, option.id));
        });
      });
      return criteria;
    }
    const fetchData = async () => {
      isLoading.value = true;
      try {
        const criteria = new useRepository.Criteria();
        criteria.setLimit(paginationLimit.value);
        criteria.setPage(currentPage.value);
        if (sortBy.value) {
          criteria.addSorting({
            field: sortBy.value,
            // TODO: Add support for checking natural sorting
            naturalSorting: false,
            order: sortDirection.value
          });
        }
        if (searchValue.value) {
          criteria.setTerm(searchValue.value);
        }
        if (appliedFilters.value.length > 0) {
          addDataTableFilters(criteria, appliedFilters.value);
        }
        const data = await repository.value.search(criteria);
        if (data) {
          dataSource.value = data;
          paginationTotalItems.value = data.total ?? 0;
          numberOfResults.value = data.total ?? 0;
        }
      } catch (error) {
        console.error(error);
      } finally {
        isLoading.value = false;
      }
    };
    vue.watch([currentPage, paginationLimit, sortBy, sortDirection, searchValue, appliedFilters], () => {
      fetchData();
    });
    vue.onMounted(() => {
      fetchData();
    });
    const handlePaginationCurrentPageChange = (page) => {
      currentPage.value = page;
    };
    const handlePaginationLimitChange = (limit) => {
      paginationLimit.value = limit;
    };
    const handleChangeEnableRowNumbering = (newEnableRowNumbering) => {
      enableRowNumbering.value = newEnableRowNumbering;
    };
    const handleChangeShowOutlines = (newShowOutlines) => {
      showOutlines.value = newShowOutlines;
    };
    const handleChangeShowStripes = (newShowStripes) => {
      showStripes.value = newShowStripes;
    };
    const handleChangeOutlineFraming = (newEnableOutlineFraming) => {
      enableOutlineFraming.value = newEnableOutlineFraming;
    };
    const handleSortChange = (newSortBy, newSortDirection) => {
      sortBy.value = newSortBy;
      sortDirection.value = newSortDirection;
    };
    const handleSearchValueChange = (newSearchValue) => {
      searchValue.value = newSearchValue;
    };
    const handleSelectionChange = (row) => {
      if (row.value && !selectedRows.value.includes(row.id)) {
        selectedRows.value.push(row.id);
      } else {
        selectedRows.value = selectedRows.value.filter((id) => id !== row.id);
      }
    };
    const handleMultipleSelectionChange = ({
      selections,
      value
    }) => {
      if (value) {
        selectedRows.value = Array.from(/* @__PURE__ */ new Set([...selectedRows.value, ...selections]));
      } else {
        selectedRows.value = selectedRows.value.filter((id) => !selections.includes(id));
      }
    };
    const handleOpenDetails = (row) => {
      emit("open-details", row);
    };
    const handleBulkDelete = () => {
      emit("bulk-delete", selectedRows.value);
    };
    const handleBulkEdit = () => {
      emit("bulk-edit", selectedRows.value);
    };
    const handleItemDelete = async ({ id }) => {
      itemToDelete.value = id;
      const isInIframe = window.self !== window.top;
      if (isInIframe && !props.forceRealModal) {
        const { ui } = await Promise.resolve().then(() => require("./index-ee78b310.js"));
        ui.modal.open({
          title: t("entity.delete.title"),
          textContent: t("entity.delete.description"),
          buttons: [
            {
              label: t("entity.delete.cancel"),
              variant: "ghost",
              size: "small",
              method: () => {
                ui.modal.close({});
              }
            },
            {
              label: t("entity.delete.delete"),
              variant: "danger",
              size: "small",
              method: () => {
                handleItemDeleteConfirm();
                ui.modal.close({});
              }
            }
          ]
        });
      } else {
        isDeleteModalOpen.value = true;
      }
    };
    const handleItemDeleteConfirm = async () => {
      try {
        isDeletingEntity.value = true;
        if (itemToDelete.value) {
          await repository.value.delete(itemToDelete.value);
        }
      } catch (error) {
        console.error(error);
      } finally {
        isDeleteModalOpen.value = false;
        isDeletingEntity.value = false;
        await fetchData();
      }
    };
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock(vue.Fragment, null, [
        vue.createVNode(MtDataTable, vue.mergeProps(_ctx.$attrs, {
          "applied-filters": appliedFilters.value,
          "onUpdate:appliedFilters": _cache[0] || (_cache[0] = ($event) => appliedFilters.value = $event),
          title: _ctx.title,
          subtitle: _ctx.subtitle,
          layout: _ctx.layout,
          columns: _ctx.columns,
          "column-changes": _ctx.columnChanges,
          "data-source": dataSource.value,
          "current-page": currentPage.value,
          "pagination-limit": paginationLimit.value,
          "pagination-total-items": paginationTotalItems.value,
          "is-loading": isLoading.value,
          "enable-row-numbering": enableRowNumbering.value,
          "show-outlines": showOutlines.value,
          "show-stripes": showStripes.value,
          "enable-outline-framing": enableOutlineFraming.value,
          "allow-bulk-delete": _ctx.allowBulkDelete,
          "allow-bulk-edit": _ctx.allowBulkEdit,
          "allow-row-selection": _ctx.allowRowSelection,
          "bulk-edit-more-actions": _ctx.bulkEditMoreActions,
          "disable-delete": _ctx.disableDelete,
          "disable-edit": _ctx.disableEdit,
          "disable-search": _ctx.disableSearch,
          "disable-settings-table": _ctx.disableSettingsTable,
          caption: _ctx.caption,
          "sort-by": sortBy.value,
          "sort-direction": sortDirection.value,
          "pagination-options": _ctx.paginationOptions,
          "enable-reload": true,
          "search-value": searchValue.value,
          "selected-rows": selectedRows.value,
          filters: vue.unref(generatedFilters),
          "number-of-results": numberOfResults.value,
          onPaginationCurrentPageChange: handlePaginationCurrentPageChange,
          onPaginationLimitChange: handlePaginationLimitChange,
          onChangeEnableRowNumbering: handleChangeEnableRowNumbering,
          onChangeShowOutlines: handleChangeShowOutlines,
          onChangeShowStripes: handleChangeShowStripes,
          onChangeOutlineFraming: handleChangeOutlineFraming,
          onSortChange: handleSortChange,
          onSearchValueChange: handleSearchValueChange,
          onSelectionChange: handleSelectionChange,
          onMultipleSelectionChange: handleMultipleSelectionChange,
          onReload: fetchData,
          onOpenDetails: handleOpenDetails,
          onBulkDelete: handleBulkDelete,
          onBulkEdit: handleBulkEdit,
          onItemDelete: handleItemDelete
        }), vue.createSlots({ _: 2 }, [
          vue.renderList(_ctx.$slots, (_, name) => {
            return {
              name,
              fn: vue.withCtx((scope) => [
                vue.renderSlot(_ctx.$slots, name, vue.normalizeProps(vue.guardReactiveProps(scope ?? {})))
              ])
            };
          })
        ]), 1040, ["applied-filters", "title", "subtitle", "layout", "columns", "column-changes", "data-source", "current-page", "pagination-limit", "pagination-total-items", "is-loading", "enable-row-numbering", "show-outlines", "show-stripes", "enable-outline-framing", "allow-bulk-delete", "allow-bulk-edit", "allow-row-selection", "bulk-edit-more-actions", "disable-delete", "disable-edit", "disable-search", "disable-settings-table", "caption", "sort-by", "sort-direction", "pagination-options", "search-value", "selected-rows", "filters", "number-of-results"]),
        vue.createVNode(MtModalRoot, {
          "is-open": isDeleteModalOpen.value,
          onChange: _cache[2] || (_cache[2] = (isOpen) => isDeleteModalOpen.value = isOpen)
        }, {
          default: vue.withCtx(() => [
            vue.createVNode(MtModal, {
              title: vue.unref(t)("entity.delete.title"),
              width: "s"
            }, {
              default: vue.withCtx(() => [
                vue.createElementVNode("p", null, vue.toDisplayString(vue.unref(t)("entity.delete.description")), 1),
                isDeletingEntity.value ? (vue.openBlock(), vue.createBlock(MtLoader, { key: 0 })) : vue.createCommentVNode("", true)
              ]),
              footer: vue.withCtx(() => [
                vue.createElementVNode("div", _hoisted_1, [
                  vue.createVNode(MtButton, {
                    variant: "secondary",
                    onClick: _cache[1] || (_cache[1] = () => isDeleteModalOpen.value = false)
                  }, {
                    default: vue.withCtx(() => [
                      vue.createTextVNode(vue.toDisplayString(vue.unref(t)("entity.delete.cancel")), 1)
                    ]),
                    _: 1
                  }),
                  vue.createVNode(MtButton, {
                    variant: "critical",
                    onClick: handleItemDeleteConfirm
                  }, {
                    default: vue.withCtx(() => [
                      vue.createTextVNode(vue.toDisplayString(vue.unref(t)("entity.delete.delete")), 1)
                    ]),
                    _: 1
                  })
                ])
              ]),
              _: 1
            }, 8, ["title"])
          ]),
          _: 1
        }, 8, ["is-open"])
      ], 64);
    };
  }
});
const mtEntityDataTable_vue_vue_type_style_index_0_lang = "";
exports._sfc_main = _sfc_main;
