// @shopware-ag/meteor-component-library/dist/mt-floating-ui.vue_vue_type_style_index_0_lang-195e2ee3.js
import './mt-floating-ui.css';
"use strict";
const vue = require("vue");
const floatingUi_dom = require("./floating-ui.dom-612a59d8.js");
function tryOnScopeDispose(fn) {
  if (vue.getCurrentScope()) {
    vue.onScopeDispose(fn);
    return true;
  }
  return false;
}
const isClient = typeof window !== "undefined" && typeof document !== "undefined";
typeof WorkerGlobalScope !== "undefined" && globalThis instanceof WorkerGlobalScope;
const toString = Object.prototype.toString;
const isObject = (val) => toString.call(val) === "[object Object]";
const noop = () => {
};
const isIOS = /* @__PURE__ */ getIsIOS();
function getIsIOS() {
  var _a, _b;
  return isClient && ((_a = window == null ? void 0 : window.navigator) == null ? void 0 : _a.userAgent) && (/iP(?:ad|hone|od)/.test(window.navigator.userAgent) || ((_b = window == null ? void 0 : window.navigator) == null ? void 0 : _b.maxTouchPoints) > 2 && /iPad|Macintosh/.test(window == null ? void 0 : window.navigator.userAgent));
}
function toArray(value) {
  return Array.isArray(value) ? value : [value];
}
function toRef(...args) {
  if (args.length !== 1)
    return vue.toRef(...args);
  const r = args[0];
  return typeof r === "function" ? vue.readonly(vue.customRef(() => ({ get: r, set: noop }))) : vue.ref(r);
}
const defaultWindow = isClient ? window : void 0;
function unrefElement(elRef) {
  var _a;
  const plain = vue.toValue(elRef);
  return (_a = plain == null ? void 0 : plain.$el) != null ? _a : plain;
}
function useEventListener(...args) {
  let target;
  let events;
  let listeners;
  let options;
  if (typeof args[0] === "string" || Array.isArray(args[0])) {
    [events, listeners, options] = args;
    target = defaultWindow;
  } else {
    [target, events, listeners, options] = args;
  }
  if (!target)
    return noop;
  events = toArray(events);
  listeners = toArray(listeners);
  const cleanups = [];
  const cleanup = () => {
    cleanups.forEach((fn) => fn());
    cleanups.length = 0;
  };
  const register = (el, event, listener, options2) => {
    el.addEventListener(event, listener, options2);
    return () => el.removeEventListener(event, listener, options2);
  };
  const stopWatch = vue.watch(
    () => [unrefElement(target), vue.toValue(options)],
    ([el, options2]) => {
      cleanup();
      if (!el)
        return;
      const optionsClone = isObject(options2) ? { ...options2 } : options2;
      cleanups.push(
        ...events.flatMap((event) => {
          return listeners.map((listener) => register(el, event, listener, optionsClone));
        })
      );
    },
    { immediate: true, flush: "post" }
  );
  const stop = () => {
    stopWatch();
    cleanup();
  };
  tryOnScopeDispose(stop);
  return stop;
}
let _iOSWorkaround = false;
function onClickOutside(target, handler, options = {}) {
  const { window: window2 = defaultWindow, ignore = [], capture = true, detectIframe = false } = options;
  if (!window2)
    return noop;
  if (isIOS && !_iOSWorkaround) {
    _iOSWorkaround = true;
    Array.from(window2.document.body.children).forEach((el) => el.addEventListener("click", noop));
    window2.document.documentElement.addEventListener("click", noop);
  }
  let shouldListen = true;
  const shouldIgnore = (event) => {
    return vue.toValue(ignore).some((target2) => {
      if (typeof target2 === "string") {
        return Array.from(window2.document.querySelectorAll(target2)).some((el) => el === event.target || event.composedPath().includes(el));
      } else {
        const el = unrefElement(target2);
        return el && (event.target === el || event.composedPath().includes(el));
      }
    });
  };
  function hasMultipleRoots(target2) {
    const vm = vue.toValue(target2);
    return vm && vm.$.subTree.shapeFlag === 16;
  }
  function checkMultipleRoots(target2, event) {
    const vm = vue.toValue(target2);
    const children = vm.$.subTree && vm.$.subTree.children;
    if (children == null || !Array.isArray(children))
      return false;
    return children.some((child) => child.el === event.target || event.composedPath().includes(child.el));
  }
  const listener = (event) => {
    const el = unrefElement(target);
    if (event.target == null)
      return;
    if (!(el instanceof Element) && hasMultipleRoots(target) && checkMultipleRoots(target, event))
      return;
    if (!el || el === event.target || event.composedPath().includes(el))
      return;
    if (event.detail === 0)
      shouldListen = !shouldIgnore(event);
    if (!shouldListen) {
      shouldListen = true;
      return;
    }
    handler(event);
  };
  let isProcessingClick = false;
  const cleanup = [
    useEventListener(window2, "click", (event) => {
      if (!isProcessingClick) {
        isProcessingClick = true;
        setTimeout(() => {
          isProcessingClick = false;
        }, 0);
        listener(event);
      }
    }, { passive: true, capture }),
    useEventListener(window2, "pointerdown", (e) => {
      const el = unrefElement(target);
      shouldListen = !shouldIgnore(e) && !!(el && !e.composedPath().includes(el));
    }, { passive: true }),
    detectIframe && useEventListener(window2, "blur", (event) => {
      setTimeout(() => {
        var _a;
        const el = unrefElement(target);
        if (((_a = window2.document.activeElement) == null ? void 0 : _a.tagName) === "IFRAME" && !(el == null ? void 0 : el.contains(window2.document.activeElement))) {
          handler(event);
        }
      }, 0);
    }, { passive: true })
  ].filter(Boolean);
  const stop = () => cleanup.forEach((fn) => fn());
  return stop;
}
const vOnClickOutside = {
  mounted(el, binding) {
    const capture = !binding.modifiers.bubble;
    if (typeof binding.value === "function") {
      el.__onClickOutside_stop = onClickOutside(el, binding.value, { capture });
    } else {
      const [handler, options] = binding.value;
      el.__onClickOutside_stop = onClickOutside(el, handler, Object.assign({ capture }, options));
    }
  },
  unmounted(el) {
    el.__onClickOutside_stop();
  }
};
function resolveElement(el) {
  if (typeof Window !== "undefined" && el instanceof Window)
    return el.document.documentElement;
  if (typeof Document !== "undefined" && el instanceof Document)
    return el.documentElement;
  return el;
}
function checkOverflowScroll(ele) {
  const style = window.getComputedStyle(ele);
  if (style.overflowX === "scroll" || style.overflowY === "scroll" || style.overflowX === "auto" && ele.clientWidth < ele.scrollWidth || style.overflowY === "auto" && ele.clientHeight < ele.scrollHeight) {
    return true;
  } else {
    const parent = ele.parentNode;
    if (!parent || parent.tagName === "BODY")
      return false;
    return checkOverflowScroll(parent);
  }
}
function preventDefault(rawEvent) {
  const e = rawEvent || window.event;
  const _target = e.target;
  if (checkOverflowScroll(_target))
    return false;
  if (e.touches.length > 1)
    return true;
  if (e.preventDefault)
    e.preventDefault();
  return false;
}
const elInitialOverflow = /* @__PURE__ */ new WeakMap();
function useScrollLock(element, initialState = false) {
  const isLocked = vue.ref(initialState);
  let stopTouchMoveListener = null;
  let initialOverflow = "";
  vue.watch(toRef(element), (el) => {
    const target = resolveElement(vue.toValue(el));
    if (target) {
      const ele = target;
      if (!elInitialOverflow.get(ele))
        elInitialOverflow.set(ele, ele.style.overflow);
      if (ele.style.overflow !== "hidden")
        initialOverflow = ele.style.overflow;
      if (ele.style.overflow === "hidden")
        return isLocked.value = true;
      if (isLocked.value)
        return ele.style.overflow = "hidden";
    }
  }, {
    immediate: true
  });
  const lock = () => {
    const el = resolveElement(vue.toValue(element));
    if (!el || isLocked.value)
      return;
    if (isIOS) {
      stopTouchMoveListener = useEventListener(
        el,
        "touchmove",
        (e) => {
          preventDefault(e);
        },
        { passive: false }
      );
    }
    el.style.overflow = "hidden";
    isLocked.value = true;
  };
  const unlock = () => {
    const el = resolveElement(vue.toValue(element));
    if (!el || !isLocked.value)
      return;
    if (isIOS)
      stopTouchMoveListener == null ? void 0 : stopTouchMoveListener();
    el.style.overflow = initialOverflow;
    elInitialOverflow.delete(el);
    isLocked.value = false;
  };
  tryOnScopeDispose(unlock);
  return vue.computed({
    get() {
      return isLocked.value;
    },
    set(v) {
      if (v)
        lock();
      else
        unlock();
    }
  });
}
function onScrollLock() {
  let isMounted = false;
  const state = vue.ref(false);
  return (el, binding) => {
    state.value = binding.value;
    if (isMounted)
      return;
    isMounted = true;
    const isLocked = useScrollLock(el, binding.value);
    vue.watch(state, (v) => isLocked.value = v);
  };
}
onScrollLock();
const _hoisted_1 = ["data-show"];
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-floating-ui",
  props: {
    isOpened: { type: Boolean },
    floatingUiOptions: {},
    showArrow: { type: Boolean },
    offset: {},
    autoUpdateOptions: {},
    matchReferenceWidth: { type: Boolean }
  },
  emits: ["close"],
  setup(__props, { emit: __emit }) {
    var _a;
    const props = __props;
    const emit = __emit;
    let floatingUiContent = vue.ref(null);
    const floatingUiTrigger = vue.ref(null);
    const floatingUiArrow = vue.ref(null);
    const floatingUi = vue.ref(null);
    const referenceElementHeight = vue.ref(0);
    const referenceElementWidth = vue.ref(0);
    let cleanup;
    const bodyContainer = window.document.querySelector("body");
    const originalParentContainer = (_a = floatingUiContent.value) == null ? void 0 : _a.parentElement;
    const contentStyles = vue.computed(() => {
      const styles = {};
      if (props.matchReferenceWidth) {
        styles.width = `${referenceElementWidth.value}px`;
      }
      return styles;
    });
    const createFloatingUi = () => {
      var _a2;
      if (!floatingUiTrigger.value || !floatingUiContent.value) {
        return;
      }
      bodyContainer.appendChild(floatingUiContent.value);
      const givenClasses = [...((_a2 = floatingUi.value) == null ? void 0 : _a2.classList.values()) ?? []].filter(
        (c) => c !== "mt-floating-ui"
      );
      floatingUiContent.value.classList.add(...givenClasses);
      cleanup = floatingUi_dom.autoUpdate(
        floatingUiTrigger.value,
        floatingUiContent.value,
        () => {
          var _a3;
          if (!floatingUiTrigger.value || !floatingUiContent.value) {
            return;
          }
          floatingUi_dom.computePosition(floatingUiTrigger.value, floatingUiContent.value, {
            placement: "bottom-start",
            strategy: "fixed",
            middleware: [
              floatingUi_dom.offset(props.offset ?? 6),
              ...(() => {
                if (props.showArrow && floatingUiArrow.value) {
                  return [floatingUi_dom.arrow({ element: floatingUiArrow.value })];
                }
                return [];
              })(),
              floatingUi_dom.flip(),
              ...((_a3 = props.floatingUiOptions) == null ? void 0 : _a3.middleware) ?? [],
              floatingUi_dom.size({
                apply({ rects }) {
                  referenceElementWidth.value = rects.reference.width ?? 0;
                  referenceElementHeight.value = rects.reference.height ?? 0;
                }
              })
            ],
            ...props.floatingUiOptions
          }).then(({ x, y, middlewareData, placement }) => {
            if (!floatingUiContent.value) {
              return;
            }
            const staticSide = {
              top: "bottom",
              right: "left",
              bottom: "top",
              left: "right"
            }[placement.split("-")[0]];
            if (props.showArrow && floatingUiArrow.value && middlewareData.arrow) {
              Object.assign(floatingUiArrow.value.style, {
                left: middlewareData.arrow.x != null ? `${middlewareData.arrow.x}px` : "",
                top: middlewareData.arrow.y != null ? `${middlewareData.arrow.y}px` : "",
                right: "",
                bottom: "",
                [staticSide]: "-2px"
              });
            }
            Object.assign(floatingUiContent.value.style, {
              left: `${x}px`,
              top: `${y}px`
            });
            ["top", "right", "bottom", "left"].forEach((side) => {
              var _a4;
              (_a4 = floatingUiContent.value) == null ? void 0 : _a4.classList.remove(`mt-floating-ui--${side}`);
            });
            floatingUiContent.value.classList.add(`mt-floating-ui--${staticSide}`);
          });
        },
        {
          // fixes endless compute loop in rare situations (e.g. data-table)
          layoutShift: false,
          ...props.autoUpdateOptions
        }
      );
    };
    const removeFloatingUi = () => {
      if (cleanup) {
        cleanup();
      }
      if (floatingUiContent.value && // floatingUiContent.value have to be direct child of bodyContainer
      floatingUiContent.value.parentElement === bodyContainer) {
        originalParentContainer == null ? void 0 : originalParentContainer.appendChild(floatingUiContent.value);
      }
    };
    vue.watch(
      () => props.isOpened,
      (isOpened) => {
        if (isOpened) {
          vue.nextTick(() => {
            createFloatingUi();
          });
        } else {
          removeFloatingUi();
        }
      },
      { immediate: true }
    );
    const onClickOutside2 = (event) => {
      var _a2;
      if ((_a2 = floatingUi.value) == null ? void 0 : _a2.contains(event.target)) {
        return;
      }
      emit("close");
    };
    vue.onBeforeUnmount(() => {
      var _a2;
      removeFloatingUi();
      if ((floatingUiContent == null ? void 0 : floatingUiContent.value) && originalParentContainer) {
        originalParentContainer == null ? void 0 : originalParentContainer.removeChild(floatingUiContent == null ? void 0 : floatingUiContent.value);
      } else {
        (_a2 = floatingUiContent == null ? void 0 : floatingUiContent.value) == null ? void 0 : _a2.remove();
      }
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", {
        ref_key: "floatingUi",
        ref: floatingUi,
        class: "mt-floating-ui"
      }, [
        vue.createElementVNode("div", {
          ref_key: "floatingUiTrigger",
          ref: floatingUiTrigger,
          class: "mt-floating-ui__trigger"
        }, [
          vue.renderSlot(_ctx.$slots, "trigger")
        ], 512),
        _ctx.isOpened ? vue.withDirectives((vue.openBlock(), vue.createElementBlock("div", {
          key: 0,
          ref_key: "floatingUiContent",
          ref: floatingUiContent,
          class: "mt-floating-ui__content",
          "data-show": _ctx.isOpened,
          tabindex: "0",
          style: vue.normalizeStyle(contentStyles.value)
        }, [
          _ctx.showArrow ? (vue.openBlock(), vue.createElementBlock("div", {
            key: 0,
            ref_key: "floatingUiArrow",
            ref: floatingUiArrow,
            class: "mt-floating-ui__arrow",
            "data-popper-arrow": ""
          }, null, 512)) : vue.createCommentVNode("", true),
          vue.createVNode(vue.Transition, { name: "popoverTransition" }, {
            default: vue.withCtx(() => [
              _ctx.isOpened ? vue.renderSlot(_ctx.$slots, "default", {
                key: 0,
                referenceElementWidth: referenceElementWidth.value,
                referenceElementHeight: referenceElementHeight.value
              }) : vue.createCommentVNode("", true)
            ]),
            _: 3
          })
        ], 12, _hoisted_1)), [
          [vue.unref(vOnClickOutside), onClickOutside2]
        ]) : vue.createCommentVNode("", true)
      ], 512);
    };
  }
});
const mtFloatingUi_vue_vue_type_style_index_0_lang = "";
exports._sfc_main = _sfc_main;
