// @shopware-ag/meteor-component-library/dist/mt-highlight-text.vue_vue_type_style_index_0_lang-550f3bfe.mjs
import './mt-highlight-text.css';
import { defineComponent, h } from "vue";
const _sfc_main = defineComponent({
  name: "MtHighlightText",
  props: {
    searchTerm: {
      type: String,
      required: false,
      default: null
    },
    text: {
      type: String,
      required: false,
      default: null
    }
  },
  methods: {
    searchAndReplace() {
      if (!this.text) {
        return "";
      }
      if (!this.searchTerm) {
        return this.text;
      }
      const prefix = '<span class="mt-highlight-text__highlight">';
      const suffix = "</span>";
      const regExp = new RegExp(this.escapeRegExp(this.searchTerm), "ig");
      return this.text.replace(regExp, (str) => `${prefix}${str}${suffix}`);
    },
    // Remove regex special characters from search string
    escapeRegExp(string) {
      return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
    }
  },
  render() {
    return h("div", {
      class: "mt-highlight-text",
      innerHTML: this.searchAndReplace()
    });
  }
});
const mtHighlightText_vue_vue_type_style_index_0_lang = "";
export {
  _sfc_main as _
};
