// @shopware-ag/meteor-component-library/dist/mt-label.vue_vue_type_style_index_0_lang-44b4947e.js
import './mt-label.css';
"use strict";
const vue = require("vue");
const mtIcon_vue_vue_type_style_index_0_lang = require("./mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
const MtColorBadge = require("./common/MtColorBadge.js");
const vueI18n = require("vue-i18n");
const _hoisted_1 = { class: "mt-label__caption" };
const _hoisted_2 = ["title"];
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-label",
  props: {
    variant: { default: "" },
    size: { default: "default" },
    appearance: { default: "default" },
    ghost: { type: Boolean, default: false },
    caps: { type: Boolean, default: false },
    dismissable: { type: Boolean }
  },
  setup(__props) {
    const props = __props;
    const { t } = vueI18n.useI18n({
      messages: {
        en: {
          remove: "Remove"
        },
        de: {
          remove: "Entfernen"
        }
      }
    });
    const attrs = vue.useAttrs();
    const showDismissable = vue.computed(() => !!attrs.onDismiss && props.dismissable);
    const labelClasses = vue.computed(() => {
      return [
        `mt-label--appearance-${props.appearance}`,
        `mt-label--size-${props.size}`,
        {
          [`mt-label--${props.variant}`]: !!props.variant,
          "mt-label--dismissable": showDismissable,
          "mt-label--ghost": props.ghost,
          "mt-label--caps": props.caps
        }
      ];
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("span", {
        class: vue.normalizeClass(["mt-label", labelClasses.value]),
        onClick: _cache[1] || (_cache[1] = vue.withModifiers(($event) => _ctx.$emit("selected"), ["stop"])),
        tabindex: "0"
      }, [
        _ctx.appearance === "badged" ? (vue.openBlock(), vue.createBlock(MtColorBadge, {
          key: 0,
          variant: _ctx.variant,
          rounded: true
        }, null, 8, ["variant"])) : vue.createCommentVNode("", true),
        vue.createElementVNode("span", _hoisted_1, [
          vue.renderSlot(_ctx.$slots, "default")
        ]),
        showDismissable.value ? (vue.openBlock(), vue.createElementBlock("button", {
          key: 1,
          class: "mt-label__dismiss",
          title: vue.unref(t)("remove"),
          onClick: _cache[0] || (_cache[0] = vue.withModifiers(($event) => _ctx.$emit("dismiss"), ["prevent", "stop"]))
        }, [
          vue.renderSlot(_ctx.$slots, "dismiss-icon", {}, () => [
            vue.createVNode(mtIcon_vue_vue_type_style_index_0_lang._sfc_main, {
              "data-testid": "dismiss-label",
              name: "regular-times-xxs",
              size: "var(--scale-size-10)"
            })
          ])
        ], 8, _hoisted_2)) : vue.createCommentVNode("", true)
      ], 2);
    };
  }
});
const mtLabel_vue_vue_type_style_index_0_lang = "";
exports._sfc_main = _sfc_main;
