// @shopware-ag/meteor-component-library/dist/mt-label.vue_vue_type_style_index_0_lang-13843d32.mjs
import './mt-label.css';
import { defineComponent, useAttrs, computed, openBlock, createElementBlock, normalizeClass, withModifiers, createBlock, createCommentVNode, createElementVNode, renderSlot, unref, createVNode } from "vue";
import { _ as _sfc_main$1 } from "./mt-icon.vue_vue_type_style_index_0_lang-2cc5f73e.mjs";
import MtColorBadge from "./esm/MtColorBadge.js";
import { useI18n } from "vue-i18n";
const _hoisted_1 = { class: "mt-label__caption" };
const _hoisted_2 = ["title"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-label",
  props: {
    variant: { default: "" },
    size: { default: "default" },
    appearance: { default: "default" },
    ghost: { type: Boolean, default: false },
    caps: { type: Boolean, default: false },
    dismissable: { type: Boolean }
  },
  setup(__props) {
    const props = __props;
    const { t } = useI18n({
      messages: {
        en: {
          remove: "Remove"
        },
        de: {
          remove: "Entfernen"
        }
      }
    });
    const attrs = useAttrs();
    const showDismissable = computed(() => !!attrs.onDismiss && props.dismissable);
    const labelClasses = computed(() => {
      return [
        `mt-label--appearance-${props.appearance}`,
        `mt-label--size-${props.size}`,
        {
          [`mt-label--${props.variant}`]: !!props.variant,
          "mt-label--dismissable": showDismissable,
          "mt-label--ghost": props.ghost,
          "mt-label--caps": props.caps
        }
      ];
    });
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("span", {
        class: normalizeClass(["mt-label", labelClasses.value]),
        onClick: _cache[1] || (_cache[1] = withModifiers(($event) => _ctx.$emit("selected"), ["stop"])),
        tabindex: "0"
      }, [
        _ctx.appearance === "badged" ? (openBlock(), createBlock(MtColorBadge, {
          key: 0,
          variant: _ctx.variant,
          rounded: true
        }, null, 8, ["variant"])) : createCommentVNode("", true),
        createElementVNode("span", _hoisted_1, [
          renderSlot(_ctx.$slots, "default")
        ]),
        showDismissable.value ? (openBlock(), createElementBlock("button", {
          key: 1,
          class: "mt-label__dismiss",
          title: unref(t)("remove"),
          onClick: _cache[0] || (_cache[0] = withModifiers(($event) => _ctx.$emit("dismiss"), ["prevent", "stop"]))
        }, [
          renderSlot(_ctx.$slots, "dismiss-icon", {}, () => [
            createVNode(_sfc_main$1, {
              "data-testid": "dismiss-label",
              name: "regular-times-xxs",
              size: "var(--scale-size-10)"
            })
          ])
        ], 8, _hoisted_2)) : createCommentVNode("", true)
      ], 2);
    };
  }
});
const mtLabel_vue_vue_type_style_index_0_lang = "";
export {
  _sfc_main as _
};
