// @shopware-ag/meteor-component-library/dist/mt-switch.vue_vue_type_style_index_0_lang-8d1c4366.js
import './mt-switch.css';
"use strict";
const vue = require("vue");
const MtFieldError = require("./common/MtFieldError.js");
const MtFieldLabel = require("./common/MtFieldLabel.js");
const MtHelpText = require("./common/MtHelpText.js");
const useFutureFlags = require("./useFutureFlags-35232480.js");
const _hoisted_1 = ["checked", "id", "disabled", "required", "aria-describedby", "aria-invalid", "aria-label", "name"];
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-switch",
  props: {
    modelValue: { type: Boolean },
    label: {},
    isInherited: { type: Boolean },
    isInheritanceField: { type: Boolean },
    inheritedValue: { type: Boolean },
    required: { type: Boolean },
    disabled: { type: Boolean },
    checked: { type: Boolean },
    bordered: { type: Boolean },
    helpText: {},
    error: {},
    removeTopMargin: { type: Boolean },
    name: {}
  },
  emits: ["change", "update:modelValue", "inheritance-remove", "inheritance-restore"],
  setup(__props) {
    const id = vue.useId();
    const errorId = vue.useId();
    const futureFlags = useFutureFlags.useFutureFlags();
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", {
        class: vue.normalizeClass([
          "mt-switch",
          {
            "mt-switch--no-top-margin": _ctx.removeTopMargin,
            "mt-switch--future-no-default-margin": !!vue.unref(futureFlags).removeDefaultMargin,
            "mt-switch--not-bordered": !_ctx.bordered,
            "mt-switch--disabled": _ctx.disabled
          }
        ])
      }, [
        vue.createElementVNode("div", {
          class: vue.normalizeClass([
            "mt-switch__block",
            {
              "mt-switch__block--bordered": _ctx.bordered,
              "mt-switch__block--errored": !!_ctx.error && !!_ctx.bordered
            }
          ])
        }, [
          vue.createElementVNode("input", {
            checked: _ctx.isInherited ? _ctx.inheritedValue : _ctx.checked || _ctx.modelValue,
            type: "checkbox",
            class: vue.normalizeClass(["mt-switch__switch", { "mt-switch__switch--errored": !!_ctx.error }]),
            id: vue.unref(id),
            disabled: _ctx.disabled || _ctx.isInherited,
            required: _ctx.required,
            "aria-describedby": !!_ctx.error ? vue.unref(errorId) : void 0,
            "aria-invalid": !!_ctx.error || void 0,
            "aria-label": _ctx.label || _ctx.$attrs["aria-label"],
            name: _ctx.name,
            onChange: _cache[0] || (_cache[0] = vue.withModifiers(
              () => {
                _ctx.$emit("change", !_ctx.checked);
                _ctx.$emit("update:modelValue", !_ctx.modelValue);
              },
              ["stop", "prevent"]
            ))
          }, null, 42, _hoisted_1),
          vue.createVNode(MtFieldLabel, {
            class: vue.normalizeClass(["mt-switch__label", { "mt-switch__label--disabled": _ctx.disabled }]),
            for: vue.unref(id),
            inheritance: !_ctx.isInheritanceField ? "none" : _ctx.isInherited ? "linked" : "unlinked",
            "has-error": !!_ctx.error,
            required: _ctx.required,
            disabled: _ctx.disabled,
            "onUpdate:inheritance": _cache[1] || (_cache[1] = ($event) => {
              if (_ctx.isInherited) {
                _ctx.$emit("inheritance-remove");
              } else {
                _ctx.$emit("inheritance-restore");
              }
            })
          }, {
            default: vue.withCtx(() => [
              vue.createTextVNode(vue.toDisplayString(_ctx.label), 1)
            ]),
            _: 1
          }, 8, ["class", "for", "inheritance", "has-error", "required", "disabled"]),
          !!_ctx.helpText ? (vue.openBlock(), vue.createBlock(MtHelpText, {
            key: 0,
            text: _ctx.helpText,
            class: "mt-switch__help-text"
          }, null, 8, ["text"])) : vue.createCommentVNode("", true)
        ], 2),
        !!_ctx.error ? (vue.openBlock(), vue.createBlock(MtFieldError, {
          key: 0,
          error: _ctx.error,
          id: vue.unref(errorId)
        }, null, 8, ["error", "id"])) : vue.createCommentVNode("", true)
      ], 2);
    };
  }
});
const mtSwitch_vue_vue_type_style_index_0_lang = "";
exports._sfc_main = _sfc_main;
