// @shopware-ag/meteor-component-library/dist/mt-switch.vue_vue_type_style_index_0_lang-e05ec27a.mjs
import './mt-switch.css';
import { defineComponent, useId, openBlock, createElementBlock, normalizeClass, unref, createElementVNode, withModifiers, createVNode, withCtx, createTextVNode, toDisplayString, createBlock, createCommentVNode } from "vue";
import MtFieldError from "./esm/MtFieldError.js";
import MtFieldLabel from "./esm/MtFieldLabel.js";
import MtHelpText from "./esm/MtHelpText.js";
import { u as useFutureFlags } from "./useFutureFlags-2be3e179.mjs";
const _hoisted_1 = ["checked", "id", "disabled", "required", "aria-describedby", "aria-invalid", "aria-label", "name"];
const _sfc_main = /* @__PURE__ */ defineComponent({
  __name: "mt-switch",
  props: {
    modelValue: { type: Boolean },
    label: {},
    isInherited: { type: Boolean },
    isInheritanceField: { type: Boolean },
    inheritedValue: { type: Boolean },
    required: { type: Boolean },
    disabled: { type: Boolean },
    checked: { type: Boolean },
    bordered: { type: Boolean },
    helpText: {},
    error: {},
    removeTopMargin: { type: Boolean },
    name: {}
  },
  emits: ["change", "update:modelValue", "inheritance-remove", "inheritance-restore"],
  setup(__props) {
    const id = useId();
    const errorId = useId();
    const futureFlags = useFutureFlags();
    return (_ctx, _cache) => {
      return openBlock(), createElementBlock("div", {
        class: normalizeClass([
          "mt-switch",
          {
            "mt-switch--no-top-margin": _ctx.removeTopMargin,
            "mt-switch--future-no-default-margin": !!unref(futureFlags).removeDefaultMargin,
            "mt-switch--not-bordered": !_ctx.bordered,
            "mt-switch--disabled": _ctx.disabled
          }
        ])
      }, [
        createElementVNode("div", {
          class: normalizeClass([
            "mt-switch__block",
            {
              "mt-switch__block--bordered": _ctx.bordered,
              "mt-switch__block--errored": !!_ctx.error && !!_ctx.bordered
            }
          ])
        }, [
          createElementVNode("input", {
            checked: _ctx.isInherited ? _ctx.inheritedValue : _ctx.checked || _ctx.modelValue,
            type: "checkbox",
            class: normalizeClass(["mt-switch__switch", { "mt-switch__switch--errored": !!_ctx.error }]),
            id: unref(id),
            disabled: _ctx.disabled || _ctx.isInherited,
            required: _ctx.required,
            "aria-describedby": !!_ctx.error ? unref(errorId) : void 0,
            "aria-invalid": !!_ctx.error || void 0,
            "aria-label": _ctx.label || _ctx.$attrs["aria-label"],
            name: _ctx.name,
            onChange: _cache[0] || (_cache[0] = withModifiers(
              () => {
                _ctx.$emit("change", !_ctx.checked);
                _ctx.$emit("update:modelValue", !_ctx.modelValue);
              },
              ["stop", "prevent"]
            ))
          }, null, 42, _hoisted_1),
          createVNode(MtFieldLabel, {
            class: normalizeClass(["mt-switch__label", { "mt-switch__label--disabled": _ctx.disabled }]),
            for: unref(id),
            inheritance: !_ctx.isInheritanceField ? "none" : _ctx.isInherited ? "linked" : "unlinked",
            "has-error": !!_ctx.error,
            required: _ctx.required,
            disabled: _ctx.disabled,
            "onUpdate:inheritance": _cache[1] || (_cache[1] = ($event) => {
              if (_ctx.isInherited) {
                _ctx.$emit("inheritance-remove");
              } else {
                _ctx.$emit("inheritance-restore");
              }
            })
          }, {
            default: withCtx(() => [
              createTextVNode(toDisplayString(_ctx.label), 1)
            ]),
            _: 1
          }, 8, ["class", "for", "inheritance", "has-error", "required", "disabled"]),
          !!_ctx.helpText ? (openBlock(), createBlock(MtHelpText, {
            key: 0,
            text: _ctx.helpText,
            class: "mt-switch__help-text"
          }, null, 8, ["text"])) : createCommentVNode("", true)
        ], 2),
        !!_ctx.error ? (openBlock(), createBlock(MtFieldError, {
          key: 0,
          error: _ctx.error,
          id: unref(errorId)
        }, null, 8, ["error", "id"])) : createCommentVNode("", true)
      ], 2);
    };
  }
});
const mtSwitch_vue_vue_type_style_index_0_lang = "";
export {
  _sfc_main as _
};
