// @shopware-ag/meteor-component-library/dist/common/index.js
import '../index.css';
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const MtDatepicker = require("./MtDatepicker.js");
const MtBanner = require("./MtBanner.js");
const MtBadge = require("./MtBadge.js");
const MtLoader = require("./MtLoader.js");
const MtProgressBar = require("./MtProgressBar.js");
const MtButton = require("./MtButton.js");
const MtChart = require("./MtChart.js");
const MtCheckbox = require("./MtCheckbox.js");
const MtColorpicker = require("./MtColorpicker.js");
const MtEmailField = require("./MtEmailField.js");
const MtLink = require("./MtLink.js");
const MtNumberField = require("./MtNumberField.js");
const MtPasswordField = require("./MtPasswordField.js");
const MtSelect = require("./MtSelect.js");
const MtSlider = require("./MtSlider.js");
const mtSwitch_vue_vue_type_style_index_0_lang = require("../mt-switch.vue_vue_type_style_index_0_lang-8d1c4366.js");
const MtTextField = require("./MtTextField.js");
const MtTextarea = require("./MtTextarea.js");
const mtIcon_vue_vue_type_style_index_0_lang = require("../mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
const MtCard = require("./MtCard.js");
const MtEmptyState = require("./MtEmptyState.js");
const MtTabs = require("./MtTabs.js");
const MtDataTable = require("./MtDataTable.js");
const MtPagination = require("./MtPagination.js");
const MtSkeletonBar = require("./MtSkeletonBar.js");
const MtToast = require("./MtToast.js");
const MtSnackbar = require("../mt-snackbar-7b6aae13.js");
const MtAvatar = require("./MtAvatar.js");
const MtPromoBadge = require("./MtPromoBadge.js");
const MtPopover = require("./MtPopover.js");
const MtPopoverItem = require("./MtPopoverItem.js");
const MtPopoverItemResult = require("../mt-popover-item-result-77871fb7.js");
const mtFloatingUi_vue_vue_type_style_index_0_lang = require("../mt-floating-ui.vue_vue_type_style_index_0_lang-195e2ee3.js");
const MtModal = require("./MtModal.js");
const MtModalRoot = require("./MtModalRoot.js");
const mtModalClose_vue_vue_type_script_setup_true_lang = require("../mt-modal-close.vue_vue_type_script_setup_true_lang-1e877a45.js");
const mtModalTrigger_vue_vue_type_script_setup_true_lang = require("../mt-modal-trigger.vue_vue_type_script_setup_true_lang-c12ee67d.js");
const mtModalAction_vue_vue_type_script_setup_true_lang = require("../mt-modal-action.vue_vue_type_script_setup_true_lang-4abf4426.js");
const MtText = require("./MtText.js");
const MtInset = require("./MtInset.js");
const mtThemeProvider_vue_vue_type_script_setup_true_lang = require("../mt-theme-provider.vue_vue_type_script_setup_true_lang-8d6da056.js");
const tooltip_directive = require("../tooltip.directive-7b51326d.js");
const debounce = require("../debounce-77fa4677.js");
const MtTooltip = require("./MtTooltip.js");
const MtTextEditor = require("./MtTextEditor.js");
const MtTextEditorToolbarButton = require("./MtTextEditorToolbarButton.js");
const MtSearch = require("./MtSearch.js");
const MtUrlField = require("./MtUrlField.js");
const mtUnitField_vue_vue_type_script_setup_true_lang = require("../mt-unit-field.vue_vue_type_script_setup_true_lang-c12cb576.js");
const mtEntityDataTable_vue_vue_type_style_index_0_lang = require("../mt-entity-data-table.vue_vue_type_style_index_0_lang-4644129f.js");
const mtEntitySelect_vue_vue_type_script_setup_true_lang = require("../mt-entity-select.vue_vue_type_script_setup_true_lang-772811ef.js");
require("vue");
require("./MtHelpText.js");
require("../_plugin-vue_export-helper-9c783a34.js");
require("./MtFieldLabel.js");
require("./MtInheritanceSwitch.js");
require("vue-i18n");
require("./MtFieldError.js");
require("../useFutureFlags-35232480.js");
require("../useIsInsideTooltip-f4674e27.js");
require("../object-deb13c0b.js");
require("../index-d39b6c26.js");
require("../index-ab705c2a.js");
require("../mt-base-field-6a3a56a0.js");
require("./MtFieldCopyable.js");
require("../id-8e80f112.js");
require("../focus-trap.esm-8d514d11.js");
require("./MtFieldAffix.js");
require("./MtSelectBase.js");
require("./MtSelectResultList.js");
require("./MtPopoverDeprecated.js");
require("../provideInjectKeys-d5eff718.js");
require("./MtSelectResult.js");
require("./MtSelectSelectionList.js");
require("../mt-label.vue_vue_type_style_index_0_lang-44b4947e.js");
require("./MtColorBadge.js");
require("../mt-highlight-text.vue_vue_type_style_index_0_lang-a463da9a.js");
require("../mt-context-button.vue_vue_type_style_index_0_lang-8afa3ed1.js");
require("./MtContextMenuItem.js");
require("./MtPriorityPlusNavigation.js");
require("./MtDataTableSettings.js");
require("./MtDataTableTextRenderer.js");
require("./MtDataTableNumberRenderer.js");
require("./MtDataTableBadgeRenderer.js");
require("./MtDataTablePriceRenderer.js");
require("./MtSegmentedControl.js");
require("./MtDataTableResetFilterButton.js");
require("./MtDataTableFilter.js");
require("./MtToastNotification.js");
require("./MtSnackbarNotification.js");
require("./MtSmoothReflow.js");
require("../_commonjsHelpers-2cbbddc8.js");
require("../floating-ui.dom-612a59d8.js");
require("../useModalContext-e62d6d51.js");
require("../floating-ui.vue-b5ae9692.js");
require("./MtTextEditorToolbar.js");
require("./MtTextEditorToolbarButtonColor.js");
require("./MtTextEditorToolbarButtonLink.js");
require("./MtTextEditorToolbarButtonTable.js");
require("./MtTextEditorDiffModal.js");
require("./MtUnitSelect.js");
require("./MtUnitSelectResult.js");
require("../useRepository-ddf4b97b.js");
function DeviceHelper() {
  this.listeners = [];
  window.addEventListener("resize", this.resize.bind(this));
}
DeviceHelper.prototype = Object.assign(DeviceHelper.prototype, {
  /**
   * Resize method which will be fired when the user resizes the browser.
   *
   * @returns {void}
   */
  resize: debounce.debounce(function debouncedResize(event) {
    this.listeners.forEach((listenerObject) => {
      listenerObject.listener.call(listenerObject.scope, event);
    });
  }, 100),
  /**
   * Registers an event register for the browser "resize" event.
   *
   * @param {Function} callback
   * @param {Any} scope
   * @param {Object} component
   * @returns {number}
   */
  onResize({ listener, scope, component }) {
    if (!scope) {
      scope = window;
    }
    this.listeners.push({ listener, scope, component });
    return this.listeners.length - 1;
  },
  removeResizeListener(component) {
    this.listeners = this.listeners.filter(
      (listenerObject) => component !== listenerObject.component
    );
    return true;
  },
  /**
   * Returns the user agent string.
   *
   * @returns {string}
   */
  getUserAgent() {
    return window.navigator.userAgent;
  },
  /**
   * Returns the current viewport with in pixels.
   * @returns {number}
   */
  getViewportWidth() {
    return window.innerWidth;
  },
  /**
   * Returns the current viewport height in pixels.
   *
   * @returns {number}
   */
  getViewportHeight() {
    return window.innerHeight;
  },
  /**
   * Returns the pixel ratio of the device as a number.
   *
   * @returns {number}
   */
  getDevicePixelRatio() {
    return window.devicePixelRatio;
  },
  /**
   * Returns the device screen width in pixels.
   *
   * @returns {number}
   */
  getScreenWidth() {
    return window.screen.width;
  },
  /**
   * Returns the device screen height in pixels.
   *
   * @returns {number}
   */
  getScreenHeight() {
    return window.screen.height;
  },
  /**
   * Returns information about the screen orientation.
   *
   * @returns {object}
   */
  getScreenOrientation() {
    return window.screen.orientation;
  },
  /**
   * Returns the current browser language as a string.
   *
   * @returns {string}
   */
  getBrowserLanguage() {
    return window.navigator.language;
  },
  /**
   * Returns the current platform (e.g. "Win32") as a string.
   *
   * @returns {string}
   */
  getPlatform() {
    return window.navigator.platform;
  },
  /**
   * Returns the system-key (e.g. "CTRL") as a string depending of the current operating system.
   *
   * @returns {string}
   */
  getSystemKey() {
    return this.getPlatform().indexOf("Mac") > -1 ? "CTRL" : "ALT";
  }
});
const DeviceHelperPlugin = {
  install(app) {
    const deviceHelper = new DeviceHelper();
    app.config.globalProperties.$device = deviceHelper;
    app.mixin({
      unmounted() {
        this.$device.removeResizeListener(this);
      }
    });
    return true;
  }
};
const DeviceHelperPlugin$1 = DeviceHelperPlugin;
const all = "";
exports.MtDatepicker = MtDatepicker;
exports.SwDatepicker = MtDatepicker;
exports.MtBanner = MtBanner;
exports.SwBanner = MtBanner;
exports.MtBadge = MtBadge;
exports.MtLoader = MtLoader;
exports.SwLoader = MtLoader;
exports.MtProgressBar = MtProgressBar;
exports.SwProgressBar = MtProgressBar;
exports.MtButton = MtButton;
exports.SwButton = MtButton;
exports.MtChart = MtChart;
exports.MtCheckbox = MtCheckbox;
exports.SwCheckbox = MtCheckbox;
exports.MtColorpicker = MtColorpicker;
exports.SwColorpicker = MtColorpicker;
exports.MtEmailField = MtEmailField;
exports.SwEmailField = MtEmailField;
exports.MtLink = MtLink;
exports.MtNumberField = MtNumberField;
exports.SwNumberField = MtNumberField;
exports.MtPasswordField = MtPasswordField;
exports.SwPasswordField = MtPasswordField;
exports.MtSelect = MtSelect;
exports.SwSelect = MtSelect;
exports.MtSlider = MtSlider;
exports.MtSwitch = mtSwitch_vue_vue_type_style_index_0_lang._sfc_main;
exports.SwSwitch = mtSwitch_vue_vue_type_style_index_0_lang._sfc_main;
exports.MtTextField = MtTextField;
exports.SwTextField = MtTextField;
exports.MtTextarea = MtTextarea;
exports.SwTextarea = MtTextarea;
exports.MtIcon = mtIcon_vue_vue_type_style_index_0_lang._sfc_main;
exports.SwIcon = mtIcon_vue_vue_type_style_index_0_lang._sfc_main;
exports.MtCard = MtCard;
exports.SwCard = MtCard;
exports.MtEmptyState = MtEmptyState;
exports.MtTabs = MtTabs;
exports.SwTabs = MtTabs;
exports.MtDataTable = MtDataTable;
exports.SwDataTable = MtDataTable;
exports.MtPagination = MtPagination;
exports.SwPagination = MtPagination;
exports.MtSkeletonBar = MtSkeletonBar;
exports.SwSkeletonBar = MtSkeletonBar;
exports.MtToast = MtToast;
exports.MtSnackbar = MtSnackbar.mtSnackbar;
exports.useSnackbar = MtSnackbar.useSnackbar;
exports.MtAvatar = MtAvatar;
exports.MtPromoBadge = MtPromoBadge;
exports.MtPopover = MtPopover;
exports.MtPopoverItem = MtPopoverItem;
exports.MtPopoverItemResult = MtPopoverItemResult.MtPopoverItemResult;
exports.MtFloatingUi = mtFloatingUi_vue_vue_type_style_index_0_lang._sfc_main;
exports.MtModal = MtModal;
exports.MtModalRoot = MtModalRoot;
exports.MtModalClose = mtModalClose_vue_vue_type_script_setup_true_lang._sfc_main;
exports.MtModalTrigger = mtModalTrigger_vue_vue_type_script_setup_true_lang._sfc_main;
exports.MtModalAction = mtModalAction_vue_vue_type_script_setup_true_lang._sfc_main;
exports.MtText = MtText;
exports.MtInset = MtInset;
exports.MtThemeProvider = mtThemeProvider_vue_vue_type_script_setup_true_lang._sfc_main;
exports.TooltipDirective = tooltip_directive.vTooltip;
exports.MtTooltip = MtTooltip;
exports.MtTextEditor = MtTextEditor;
exports.MtTextEditorToolbarButton = MtTextEditorToolbarButton;
exports.MtSearch = MtSearch;
exports.MtUrlField = MtUrlField;
exports.MtUnitField = mtUnitField_vue_vue_type_script_setup_true_lang._sfc_main;
exports.MtEntityDataTable = mtEntityDataTable_vue_vue_type_style_index_0_lang._sfc_main;
exports.MtEntitySelect = mtEntitySelect_vue_vue_type_script_setup_true_lang._sfc_main;
exports.DeviceHelperPlugin = DeviceHelperPlugin$1;
