// @shopware-ag/meteor-component-library/dist/common/MtContextMenuItem.js
"use strict";
const MtPopoverItem = require("./MtPopoverItem.js");
const vue = require("vue");
require("vue-i18n");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
require("./MtCheckbox.js");
require("../mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
require("../mt-base-field-6a3a56a0.js");
require("./MtInheritanceSwitch.js");
require("./MtTooltip.js");
require("../floating-ui.vue-b5ae9692.js");
require("../floating-ui.dom-612a59d8.js");
require("../useIsInsideTooltip-f4674e27.js");
require("../index-ab705c2a.js");
require("./MtFieldCopyable.js");
require("../tooltip.directive-7b51326d.js");
require("../id-8e80f112.js");
require("./MtHelpText.js");
require("../useFutureFlags-35232480.js");
require("./MtFieldError.js");
require("./MtText.js");
require("../mt-switch.vue_vue_type_style_index_0_lang-8d1c4366.js");
require("./MtFieldLabel.js");
const _sfc_main = vue.defineComponent({
  name: "MtContextMenuItem",
  components: {
    "mt-popover-item": MtPopoverItem
  },
  props: {
    label: {
      type: String,
      required: true
    },
    icon: {
      type: String,
      required: false,
      default: null
    },
    disabled: {
      type: Boolean,
      required: false,
      default: false
    },
    type: {
      type: String,
      required: false,
      default: "default"
    },
    role: {
      type: String,
      required: false,
      default: "menuitem"
    }
  },
  emits: ["click"],
  setup(props, { emit }) {
    const handleLabelClick = (e) => {
      if (props.disabled) {
        return;
      }
      emit("click", e);
    };
    return {
      handleLabelClick
    };
  }
});
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  const _component_mt_popover_item = vue.resolveComponent("mt-popover-item");
  return vue.openBlock(), vue.createBlock(_component_mt_popover_item, {
    class: "mt-context-menu-item",
    label: _ctx.label,
    icon: _ctx.icon,
    disabled: _ctx.disabled,
    "on-label-click": _ctx.handleLabelClick,
    type: _ctx.type,
    role: _ctx.role
  }, null, 8, ["label", "icon", "disabled", "on-label-click", "type", "role"]);
}
const MtContextMenuItem = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["render", _sfc_render]]);
module.exports = MtContextMenuItem;
