// @shopware-ag/meteor-component-library/dist/common/MtDataTableResetFilterButton.js
import '../mt-data-table-reset-filter-button.css';
"use strict";
const vue = require("vue");
const vueI18n = require("vue-i18n");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
const _hoisted_1 = {
  type: "button",
  class: "mt-data-table-reset-filter-button"
};
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-data-table-reset-filter-button",
  props: {
    numberOfAppliedFilters: {}
  },
  setup(__props) {
    const { t } = vueI18n.useI18n({
      messages: {
        en: {
          label: "Remove filter | Remove filters"
        },
        de: {
          label: "Filter entfernen | Filter entfernen"
        }
      }
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("button", _hoisted_1, vue.toDisplayString(vue.unref(t)("label", { n: _ctx.numberOfAppliedFilters })), 1);
    };
  }
});
const mtDataTableResetFilterButton_vue_vue_type_style_index_0_scoped_ea9bb72d_lang = "";
const MtDataTableResetFilterButton = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["__scopeId", "data-v-ea9bb72d"]]);
module.exports = MtDataTableResetFilterButton;
