// @shopware-ag/meteor-component-library/dist/common/MtDataTableNumberRenderer.js
import '../mt-data-table-number-renderer.css';
"use strict";
const vue = require("vue");
const object = require("../object-deb13c0b.js");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
const _sfc_main = vue.defineComponent({
  name: "MtDataTableNumberRenderer",
  props: {
    columnDefinition: {
      type: Object,
      required: true
    },
    data: {
      type: void 0,
      required: true
    }
  },
  setup(props) {
    const renderString = vue.computed(() => {
      return String(Number(object.get(props.data, props.columnDefinition.property)));
    });
    return {
      renderString
    };
  }
});
const mtDataTableNumberRenderer_vue_vue_type_style_index_0_lang = "";
const _hoisted_1 = {
  key: 1,
  class: "mt-data-table-number-renderer"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return _ctx.columnDefinition.clickable ? (vue.openBlock(), vue.createElementBlock("a", {
    key: 0,
    class: "mt-data-table-number-renderer",
    href: "#",
    onClick: _cache[0] || (_cache[0] = vue.withModifiers(($event) => _ctx.$emit("click", _ctx.data), ["prevent"]))
  }, vue.toDisplayString(_ctx.renderString), 1)) : (vue.openBlock(), vue.createElementBlock("p", _hoisted_1, vue.toDisplayString(_ctx.renderString), 1));
}
const MtDataTableNumberRenderer = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["render", _sfc_render]]);
module.exports = MtDataTableNumberRenderer;
