// @shopware-ag/meteor-component-library/dist/common/MtDataTableFilter.js
import '../mt-data-table-filter.css';
"use strict";
const vue = require("vue");
const mtIcon_vue_vue_type_style_index_0_lang = require("../mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
const MtPopover = require("./MtPopover.js");
const MtPopoverItem = require("./MtPopoverItem.js");
const MtText = require("./MtText.js");
const vueI18n = require("vue-i18n");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
require("./MtCheckbox.js");
require("../mt-base-field-6a3a56a0.js");
require("./MtInheritanceSwitch.js");
require("./MtTooltip.js");
require("../floating-ui.vue-b5ae9692.js");
require("../floating-ui.dom-612a59d8.js");
require("../useIsInsideTooltip-f4674e27.js");
require("../index-ab705c2a.js");
require("./MtFieldCopyable.js");
require("../tooltip.directive-7b51326d.js");
require("../id-8e80f112.js");
require("./MtHelpText.js");
require("../useFutureFlags-35232480.js");
require("./MtFieldError.js");
require("../mt-switch.vue_vue_type_style_index_0_lang-8d1c4366.js");
require("./MtFieldLabel.js");
require("./MtButton.js");
require("./MtLoader.js");
require("./MtSmoothReflow.js");
require("../_commonjsHelpers-2cbbddc8.js");
require("../mt-floating-ui.vue_vue_type_style_index_0_lang-195e2ee3.js");
const _hoisted_1 = {
  class: "mt-data-table-filter",
  "data-testid": "mt-data-table-filter"
};
const _hoisted_2 = ["onClick"];
const _hoisted_3 = ["aria-label"];
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-data-table-filter",
  props: {
    filter: {},
    appliedOptions: {}
  },
  emits: ["removeOption", "addOption", "removeFilter"],
  setup(__props) {
    const props = __props;
    function isOptionSelected(optionId) {
      return !!props.appliedOptions.find((option) => option.id === optionId);
    }
    const { t } = vueI18n.useI18n({
      messages: {
        en: {
          is: "is",
          removeButton: "Remove filter"
        },
        de: {
          is: "ist",
          removeButton: "Filter entfernen"
        }
      }
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", _hoisted_1, [
        vue.createVNode(MtText, {
          class: "mt-data-table-filter__property",
          size: "2xs"
        }, {
          default: vue.withCtx(() => [
            vue.createTextVNode(vue.toDisplayString(_ctx.filter.label), 1)
          ]),
          _: 1
        }),
        vue.createVNode(MtText, {
          size: "2xs",
          color: "color-text-secondary-default",
          class: "mt-data-table-filter__rule"
        }, {
          default: vue.withCtx(() => [
            vue.createTextVNode(vue.toDisplayString(vue.unref(t)("is")), 1)
          ]),
          _: 1
        }),
        vue.createVNode(MtPopover, {
          class: "mt-data-table-filter__option",
          title: "Manufactuer"
        }, {
          trigger: vue.withCtx(({ toggleFloatingUi }) => [
            vue.createElementVNode("button", {
              type: "button",
              onKeydown: _cache[0] || (_cache[0] = vue.withKeys(($event) => _ctx.$emit("removeFilter"), ["delete"])),
              onClick: toggleFloatingUi
            }, vue.toDisplayString(_ctx.appliedOptions.map((option) => option.label).join(", ")), 41, _hoisted_2)
          ]),
          "popover-items__base": vue.withCtx(() => [
            (vue.openBlock(true), vue.createElementBlock(vue.Fragment, null, vue.renderList(_ctx.filter.type.options, (option) => {
              return vue.openBlock(), vue.createBlock(MtPopoverItem, {
                key: option.id,
                label: option.label,
                "show-checkbox": "",
                "checkbox-checked": isOptionSelected(option.id),
                onChangeCheckbox: (isChecked) => isChecked ? _ctx.$emit("addOption", _ctx.filter.id, option.id) : _ctx.$emit("removeOption", _ctx.filter.id, option.id)
              }, null, 8, ["label", "checkbox-checked", "onChangeCheckbox"]);
            }), 128))
          ]),
          _: 1
        }),
        vue.createElementVNode("button", {
          type: "button",
          class: "mt-data-table-filter__remove-button",
          "aria-label": vue.unref(t)("removeButton"),
          onKeydown: _cache[1] || (_cache[1] = vue.withKeys(($event) => _ctx.$emit("removeFilter"), ["delete"])),
          onClick: _cache[2] || (_cache[2] = ($event) => _ctx.$emit("removeFilter"))
        }, [
          vue.createVNode(mtIcon_vue_vue_type_style_index_0_lang._sfc_main, {
            name: "solid-times-xxs",
            "aria-hidden": "true"
          })
        ], 40, _hoisted_3)
      ]);
    };
  }
});
const mtDataTableFilter_vue_vue_type_style_index_0_scoped_afdb8a25_lang = "";
const MtDataTableFilter = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["__scopeId", "data-v-afdb8a25"]]);
module.exports = MtDataTableFilter;
