// @shopware-ag/meteor-component-library/dist/common/MtDataTableTextRenderer.js
import '../mt-data-table-text-renderer.css';
"use strict";
const vue = require("vue");
const object = require("../object-deb13c0b.js");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
const _sfc_main = vue.defineComponent({
  name: "MtDataTableTextRenderer",
  props: {
    columnDefinition: {
      type: Object,
      required: true
    },
    data: {
      type: void 0,
      required: true
    }
  },
  setup(props) {
    const renderPreviewImage = vue.computed(() => {
      return object.get(props.data, props.columnDefinition.previewImage || "");
    });
    const renderString = vue.computed(() => {
      return object.get(props.data, props.columnDefinition.property);
    });
    return {
      renderString,
      renderPreviewImage
    };
  }
});
const mtDataTableTextRenderer_vue_vue_type_style_index_0_scoped_745a4686_lang = "";
const _hoisted_1 = { class: "mt-data-table-text-renderer-cell" };
const _hoisted_2 = {
  key: 0,
  class: "mt-data-table-preview-image-renderer"
};
const _hoisted_3 = ["src", "alt"];
const _hoisted_4 = {
  key: 2,
  class: "mt-data-table-text-renderer"
};
function _sfc_render(_ctx, _cache, $props, $setup, $data, $options) {
  return vue.openBlock(), vue.createElementBlock("div", _hoisted_1, [
    _ctx.columnDefinition.previewImage ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_2, [
      _ctx.columnDefinition.previewImage ? (vue.openBlock(), vue.createElementBlock("img", {
        key: 0,
        class: "mt-data-table-preview-image-renderer-item",
        src: _ctx.renderPreviewImage,
        alt: _ctx.renderString
      }, null, 8, _hoisted_3)) : vue.createCommentVNode("", true)
    ])) : vue.createCommentVNode("", true),
    _ctx.columnDefinition.clickable ? (vue.openBlock(), vue.createElementBlock("a", {
      key: 1,
      class: "mt-data-table-text-renderer",
      href: "#",
      onClick: _cache[0] || (_cache[0] = vue.withModifiers(($event) => _ctx.$emit("click", _ctx.data), ["prevent"]))
    }, vue.toDisplayString(_ctx.renderString), 1)) : (vue.openBlock(), vue.createElementBlock("p", _hoisted_4, vue.toDisplayString(_ctx.renderString), 1))
  ]);
}
const MtDataTableTextRenderer = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["render", _sfc_render], ["__scopeId", "data-v-745a4686"]]);
module.exports = MtDataTableTextRenderer;
