// @shopware-ag/meteor-component-library/dist/common/MtHelpText.js
import '../mt-help-text.css';
"use strict";
const vue = require("vue");
const mtIcon_vue_vue_type_style_index_0_lang = require("../mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
const MtTooltip = require("./MtTooltip.js");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
require("../floating-ui.vue-b5ae9692.js");
require("../floating-ui.dom-612a59d8.js");
require("../useIsInsideTooltip-f4674e27.js");
require("../index-ab705c2a.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-help-text",
  props: {
    text: {},
    width: {},
    showDelay: { default: 0 },
    hideDelay: { default: 150 },
    placement: {}
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createBlock(MtTooltip, {
        content: _ctx.text,
        placement: _ctx.placement,
        "max-width": _ctx.width,
        "delay-duration-in-ms": _ctx.showDelay,
        "hide-delay-duration-in-ms": _ctx.hideDelay
      }, {
        default: vue.withCtx((props) => [
          vue.createElementVNode("button", vue.mergeProps({ type: "button" }, { ...props, ..._ctx.$attrs }, { class: "mt-help-text" }), [
            vue.createVNode(mtIcon_vue_vue_type_style_index_0_lang._sfc_main, {
              "aria-hidden": "true",
              "data-testid": "mt-help-text__icon",
              name: "solid-question-circle-s"
            })
          ], 16)
        ]),
        _: 1
      }, 8, ["content", "placement", "max-width", "delay-duration-in-ms", "hide-delay-duration-in-ms"]);
    };
  }
});
const mtHelpText_vue_vue_type_style_index_0_scoped_9f6b1469_lang = "";
const MtHelpText = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["__scopeId", "data-v-9f6b1469"]]);
module.exports = MtHelpText;
