// @shopware-ag/meteor-component-library/dist/common/MtText.js
import '../mt-text.css';
"use strict";
const vue = require("vue");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-text",
  props: {
    size: { default: "s" },
    weight: { default: "regular" },
    color: { default: "color-text-primary-default" },
    as: { default: "p" }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createBlock(vue.resolveDynamicComponent(_ctx.as), {
        class: vue.normalizeClass([`mt-text--size-${_ctx.size}`, `mt-text--weight-${_ctx.weight}`]),
        style: vue.normalizeStyle({ color: `var(--${_ctx.color})` })
      }, {
        default: vue.withCtx(() => [
          vue.renderSlot(_ctx.$slots, "default", {}, void 0, true)
        ]),
        _: 3
      }, 8, ["class", "style"]);
    };
  }
});
const mtText_vue_vue_type_style_index_0_scoped_44e2f2b3_lang = "";
const MtText = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["__scopeId", "data-v-44e2f2b3"]]);
module.exports = MtText;
