// @shopware-ag/meteor-component-library/dist/common/MtTextEditorToolbarButtonColor.js
import '../mt-text-editor-toolbar-button-color.css';
"use strict";
Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: "Module" } });
const vue = require("vue");
const MtColorpicker = require("./MtColorpicker.js");
const vueI18n = require("vue-i18n");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
require("../debounce-77fa4677.js");
require("../mt-base-field-6a3a56a0.js");
require("./MtInheritanceSwitch.js");
require("../mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
require("./MtTooltip.js");
require("../floating-ui.vue-b5ae9692.js");
require("../floating-ui.dom-612a59d8.js");
require("../useIsInsideTooltip-f4674e27.js");
require("../index-ab705c2a.js");
require("./MtFieldCopyable.js");
require("../tooltip.directive-7b51326d.js");
require("../id-8e80f112.js");
require("./MtHelpText.js");
require("../useFutureFlags-35232480.js");
require("../mt-floating-ui.vue_vue_type_style_index_0_lang-195e2ee3.js");
require("./MtText.js");
require("../focus-trap.esm-8d514d11.js");
require("./MtButton.js");
require("./MtLoader.js");
require("./MtFieldError.js");
const _hoisted_1 = { class: "mt-text-editor-toolbar-button-color" };
const colorButton = {
  name: "text-color",
  label: "mt-text-editor-toolbar-button-color.label",
  position: 2e3
};
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-text-editor-toolbar-button-color",
  props: {
    editor: {
      type: Object,
      required: true
    },
    button: {
      type: Object,
      required: true
    },
    disabled: {
      type: Boolean,
      default: false
    }
  },
  setup(__props) {
    vueI18n.useI18n({
      useScope: "global",
      messages: {
        en: {
          "mt-text-editor-toolbar-button-color": {
            label: "Text Color"
          }
        },
        de: {
          "mt-text-editor-toolbar-button-color": {
            label: "Textfarbe"
          }
        }
      }
    });
    const props = __props;
    const getTextColor = () => {
      return props.editor.getAttributes("textStyle").color;
    };
    const onUpdateModelValue = (color) => {
      props.editor.chain().focus().setColor(color).run();
    };
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", _hoisted_1, [
        vue.createVNode(MtColorpicker, {
          class: "mt-text-editor-toolbar-button-color__colorpicker",
          compact: "",
          "model-value": getTextColor(),
          "onUpdate:modelValue": onUpdateModelValue,
          disabled: __props.button.disabled ? __props.button.disabled(props.editor, __props.disabled) : __props.disabled,
          "apply-mode": ""
        }, null, 8, ["model-value", "disabled"])
      ]);
    };
  }
});
const mtTextEditorToolbarButtonColor_vue_vue_type_style_index_0_scoped_356322ec_lang = "";
const mtTextEditorToolbarButtonColor = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["__scopeId", "data-v-356322ec"]]);
exports.colorButton = colorButton;
exports.default = mtTextEditorToolbarButtonColor;
