// @shopware-ag/meteor-component-library/dist/common/MtUnitSelectResult.js
import '../mt-unit-select-result.css';
"use strict";
const vue = require("vue");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
const _hoisted_1 = {
  role: "option",
  "aria-selected": false
};
const _hoisted_2 = ["aria-label", "data-testid"];
const _hoisted_3 = { class: "mt-unit-select-result__label" };
const _hoisted_4 = { class: "mt-unit-select-result__unit" };
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-unit-select-result",
  props: {
    unit: {},
    unitLabel: {}
  },
  emits: ["click"],
  setup(__props, { emit: __emit }) {
    const emit = __emit;
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("li", _hoisted_1, [
        vue.createElementVNode("button", {
          class: "mt-unit-select-result",
          onClick: _cache[0] || (_cache[0] = ($event) => emit("click", _ctx.unit)),
          type: "button",
          "aria-label": `Select ${_ctx.unitLabel.plural} (${_ctx.unit})`,
          "data-testid": `unit-select-option-${_ctx.unit}`
        }, [
          vue.createElementVNode("span", _hoisted_3, vue.toDisplayString(_ctx.unitLabel.plural), 1),
          vue.createElementVNode("span", _hoisted_4, vue.toDisplayString(_ctx.unit), 1)
        ], 8, _hoisted_2)
      ]);
    };
  }
});
const mtUnitSelectResult_vue_vue_type_style_index_0_scoped_bbacdfea_lang = "";
const MtUnitSelectResult = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["__scopeId", "data-v-bbacdfea"]]);
module.exports = MtUnitSelectResult;
