// @shopware-ag/meteor-component-library/dist/common/MtUrlField.js
import '../mt-url-field.css';
"use strict";
const vue = require("vue");
const mtIcon_vue_vue_type_style_index_0_lang = require("../mt-icon.vue_vue_type_style_index_0_lang-0a28c7b6.js");
const MtFieldLabel = require("./MtFieldLabel.js");
const MtHelpText = require("./MtHelpText.js");
const MtFieldError = require("./MtFieldError.js");
const MtTooltip = require("./MtTooltip.js");
const index = require("../index-d39b6c26.js");
const vueI18n = require("vue-i18n");
const useFutureFlags = require("../useFutureFlags-35232480.js");
const _pluginVue_exportHelper = require("../_plugin-vue_export-helper-9c783a34.js");
require("./MtInheritanceSwitch.js");
require("./MtText.js");
require("../floating-ui.vue-b5ae9692.js");
require("../floating-ui.dom-612a59d8.js");
require("../useIsInsideTooltip-f4674e27.js");
require("../index-ab705c2a.js");
const _hoisted_1 = ["disabled", "data-size"];
const _hoisted_2 = ["id", "value", "placeholder", "name", "required", "disabled"];
const _hoisted_3 = ["aria-label"];
const _hoisted_4 = {
  key: 1,
  class: "mt-url-field__affix mt-url-field__affix--suffix"
};
const _hoisted_5 = {
  key: 2,
  class: "mt-url-field__hint"
};
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "mt-url-field",
  props: /* @__PURE__ */ vue.mergeModels({
    omitUrlHash: { type: Boolean },
    omitUrlSearch: { type: Boolean },
    copyable: { type: Boolean },
    error: {},
    label: {},
    required: { type: Boolean },
    isInheritanceField: { type: Boolean },
    isInherited: { type: Boolean },
    helpText: {},
    disabled: { type: Boolean },
    placeholder: {},
    name: {},
    size: { default: "default" }
  }, {
    "modelValue": {
      type: String
    },
    "modelModifiers": {}
  }),
  emits: /* @__PURE__ */ vue.mergeModels(["change", "inheritance-remove", "inheritance-restore"], ["update:modelValue"]),
  setup(__props) {
    const futureFlags = useFutureFlags.useFutureFlags();
    const URL_REGEX = {
      PROTOCOL: /([a-zA-Z0-9]+:\/\/)+/,
      PROTOCOL_HTTP: /^https?:\/\//,
      SSL: /^\s*https:\/\//,
      TRAILING_SLASH: /\/+$/
    };
    const modelValue = vue.useModel(__props, "modelValue");
    const props = __props;
    const id = vue.useId();
    const currentValue = vue.ref(modelValue.value);
    const sslActive = vue.ref(true);
    const urlPrefix = vue.computed(() => {
      return sslActive.value ? "https://" : "http://";
    });
    const url = vue.computed(() => {
      var _a;
      const trimmedValue = (_a = currentValue.value) == null ? void 0 : _a.trim();
      if (!trimmedValue)
        return "";
      return `${urlPrefix.value}${trimmedValue}`;
    });
    vue.watch(
      () => modelValue.value,
      () => {
        const result = checkInput(modelValue.value || "");
        currentValue.value = result;
      },
      { immediate: true }
    );
    function checkInput(inputValue) {
      if (!inputValue.length)
        return "";
      if (inputValue.match(URL_REGEX.PROTOCOL_HTTP)) {
        sslActive.value = !!inputValue.match(URL_REGEX.SSL);
      }
      const validated = transformURL(inputValue);
      if (!validated) {
        throw new Error("Invalid URL");
      } else {
        return validated;
      }
    }
    function transformURL(value) {
      const url2 = new URL(value.match(URL_REGEX.PROTOCOL) ? value : `${urlPrefix.value}${value}`);
      if (!url2)
        return null;
      if (props.omitUrlSearch)
        url2.search = "";
      if (props.omitUrlHash)
        url2.hash = "";
      const removeTrailingSlash = url2.hash === "" && url2.search === "" ? URL_REGEX.TRAILING_SLASH : "";
      return url2.toString().replace(URL_REGEX.PROTOCOL, "").replace(removeTrailingSlash, "").replace(url2.host, decodeURI(url2.host));
    }
    const { copy, copied } = index.useClipboard();
    const { t } = vueI18n.useI18n({
      messages: {
        de: {
          copyTooltip: "URL in Zwischenablage kopieren",
          copyButtonDescription: "URL in Zwischenablage kopieren",
          copyButtonDescriptionValueCopied: "URL in Zwischenablage kopiert"
        },
        en: {
          copyTooltip: "Copy URL to clipboard",
          copyButtonDescription: "Copy URL to clipboard",
          copyButtonDescriptionValueCopied: "Copied URL to clipboard"
        }
      }
    });
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", {
        class: vue.normalizeClass([
          "mt-url-field",
          {
            "mt-url-field--future-remove-default-margin": vue.unref(futureFlags).removeDefaultMargin
          }
        ])
      }, [
        !!_ctx.label ? (vue.openBlock(), vue.createBlock(MtFieldLabel, {
          key: 0,
          for: vue.unref(id),
          style: { marginBottom: "var(--scale-size-2)", gridArea: "label" },
          "has-error": !!_ctx.error,
          required: _ctx.required,
          inheritance: !_ctx.isInheritanceField ? "none" : _ctx.isInherited ? "linked" : "unlinked",
          "onUpdate:inheritance": _cache[0] || (_cache[0] = ($event) => {
            if (_ctx.isInherited) {
              _ctx.$emit("inheritance-remove");
            } else {
              _ctx.$emit("inheritance-restore");
            }
          })
        }, {
          default: vue.withCtx(() => [
            vue.createTextVNode(vue.toDisplayString(_ctx.label), 1)
          ]),
          _: 1
        }, 8, ["for", "has-error", "required", "inheritance"])) : vue.createCommentVNode("", true),
        !!_ctx.helpText ? (vue.openBlock(), vue.createBlock(MtHelpText, {
          key: 1,
          text: _ctx.helpText,
          style: { gridArea: "help-text" }
        }, null, 8, ["text"])) : vue.createCommentVNode("", true),
        vue.createElementVNode("div", {
          class: vue.normalizeClass([
            "mt-url-field__block",
            `mt-url-field__block--size-${_ctx.size}`,
            {
              "mt-url-field__block--error": !!_ctx.error
            }
          ])
        }, [
          vue.createElementVNode("button", {
            type: "button",
            class: "mt-url-field__protocol-toggle",
            disabled: _ctx.disabled || _ctx.isInherited,
            "data-size": _ctx.size,
            onClick: _cache[1] || (_cache[1] = () => {
              sslActive.value = !sslActive.value;
              modelValue.value = url.value;
            })
          }, [
            vue.createVNode(mtIcon_vue_vue_type_style_index_0_lang._sfc_main, {
              name: sslActive.value ? "solid-lock" : "solid-lock-open",
              color: sslActive.value ? "var(--color-icon-positive-default)" : "var(--color-icon-primary-default)",
              size: "var(--scale-size-12)",
              "aria-hidden": "true"
            }, null, 8, ["name", "color"]),
            vue.createElementVNode("span", null, vue.toDisplayString(urlPrefix.value), 1)
          ], 8, _hoisted_1),
          vue.createElementVNode("input", {
            id: vue.unref(id),
            value: decodeURI(currentValue.value || ""),
            type: "url",
            placeholder: _ctx.placeholder,
            name: _ctx.name,
            required: _ctx.required,
            disabled: _ctx.disabled || _ctx.isInherited,
            class: "mt-url-field__input",
            onInput: _cache[2] || (_cache[2] = (event) => {
              const result = checkInput(event.target.value);
              currentValue.value = result;
              modelValue.value = url.value;
            }),
            onChange: _cache[3] || (_cache[3] = vue.withModifiers(($event) => _ctx.$emit("change", $event.target.value || ""), ["stop"]))
          }, null, 40, _hoisted_2),
          _ctx.copyable ? (vue.openBlock(), vue.createBlock(MtTooltip, {
            key: 0,
            content: vue.unref(t)("copyTooltip")
          }, {
            default: vue.withCtx((params) => [
              vue.createElementVNode("button", vue.mergeProps({ type: "button" }, params, {
                class: "mt-url-field__copy-button",
                "aria-label": vue.unref(copied) ? vue.unref(t)("copyButtonDescriptionValueCopied") : vue.unref(t)("copyButtonDescription"),
                onClick: _cache[4] || (_cache[4] = () => {
                  if (!currentValue.value)
                    return;
                  vue.unref(copy)(currentValue.value);
                })
              }), [
                vue.createVNode(mtIcon_vue_vue_type_style_index_0_lang._sfc_main, {
                  name: vue.unref(copied) ? "regular-checkmark" : "regular-copy",
                  size: "var(--scale-size-18)",
                  color: "var(--color-icon-primary-default)"
                }, null, 8, ["name"])
              ], 16, _hoisted_3)
            ]),
            _: 1
          }, 8, ["content"])) : vue.createCommentVNode("", true),
          _ctx.$slots.suffix ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_4, [
            vue.renderSlot(_ctx.$slots, "suffix", {}, void 0, true)
          ])) : vue.createCommentVNode("", true)
        ], 2),
        vue.createVNode(MtFieldError, {
          error: _ctx.error,
          style: { gridArea: "error" }
        }, null, 8, ["error"]),
        _ctx.$slots.hint ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_5, [
          vue.renderSlot(_ctx.$slots, "hint", {}, void 0, true)
        ])) : vue.createCommentVNode("", true)
      ], 2);
    };
  }
});
const mtUrlField_vue_vue_type_style_index_0_scoped_7594679e_lang = "";
const mtUrlField = /* @__PURE__ */ _pluginVue_exportHelper._export_sfc(_sfc_main, [["__scopeId", "data-v-7594679e"]]);
module.exports = mtUrlField;
